diff -r -u cash.orig/mebdfi.f cash/mebdfi.f
--- cash.orig/mebdfi.f	2007-11-16 19:00:20.000000000 +0100
+++ cash/mebdfi.f	2007-11-28 20:26:31.000000000 +0100
@@ -541,7 +541,7 @@
          IF (IDID.NE.1) THEN
 C           VALUE OF IDID NOT ALLOWED
             WRITE (LOUT,9070) IDID
-            IDID = -4
+            IDID = -41
          END IF
 
          NN=N
@@ -550,7 +550,7 @@
             IF (RTOL(I).LT.0.0D+0) THEN
 C           ILLEGAL VALUE FOR RELATIVE ERROR TOLERENCE
                WRITE (LOUT,9040)
-               IDID = -4
+               IDID = -43
             END IF
  1       CONTINUE
 
@@ -560,13 +560,13 @@
             IF (ATOL(I).LT.0.0D+0) THEN
 C           ILLEGAL ABSOLUTE ERROR TOLERANCE
                WRITE(LOUT,9045)
-               IDID=-4
+               IDID = -42
             ENDIF
  2       CONTINUE
          IF(ITOL.EQ.1.AND.RTOL(1).EQ.0) THEN
 C           ILLEGAL ERROR TOLERANCE
             WRITE(LOUT,9040)
-            IDID = -4
+            IDID = -43
          ENDIF
          IF(ITOL.NE.1) THEN
             VHOLD = 0.0D+0
@@ -582,7 +582,7 @@
                ENDIF
                IF(VHOLD.LE.0.0D+0) THEN
                   WRITE(LOUT,9040)
-                  IDID = -4
+                  IDID = -43
                ENDIF
  3          CONTINUE
          ENDIF
@@ -590,39 +590,39 @@
          IF (N.LE.0) THEN
 C           ILLEGAL VALUE FOR THE NUMBER OF EQUATIONS
             WRITE (LOUT,9050)
-            IDID = -4
+            IDID = -44
          END IF
 
          IF ((T0-TOUT)*HO.GE.0.0D+0) THEN
 C           PARAMETERS FOR INTEGRATION ARE ILLEGAL
             WRITE (LOUT,9060)
-            IDID = -4
+            IDID = -45
          END IF
 
          IF ((MF.NE.21).AND.(MF.NE.22).AND.(MF.NE.23).AND.(MF.NE.24))
      +   THEN
 C           ILLEGAL VALUE FOR METHOD FLAG
             WRITE (LOUT,9090) MF
-            IDID = -4
+            IDID = -46
          END IF
 
          IF(ITOL.LT.1.OR.ITOL.GT.5) THEN
 C           ILLEGAL VALUE FOR ERROR CONTROL PARAMETER
             WRITE (LOUT,9110)
-            IDID=-4
+            IDID = -47
          ENDIF
 
          IF(MAXDER.LT.1.OR.MAXDER.GT.7) THEN
 C        ILLEGAL VALUE FOR MAXIMUM ORDER
             WRITE(LOUT,9120)
-            IDID = -4
+            IDID = -48
          ENDIF
 
          IF(NIND1.EQ.0)  NIND1=N
          IF(NIND1 + NIND2 + NIND3 .NE. N) THEN
 C    SUM OF VARIABLES OF DIFFERENT INDEX SHOULD BE N.
             WRITE(LOUT,9140)
-            IDID = -4
+            IDID = -49
          ENDIF
 
          IF (IDID.NE.1) THEN
Only in cash: mebdfi.f~
Only in cash: mebdfi.o
