diff -urN -x *.exe -x *.dll -x *.o -x *.a -x *.bak qhull-2003.1-orig/src/Makefile qhull-2003.1/src/Makefile
--- qhull-2003.1-orig/src/Makefile	Wed Dec 31 04:20:50 2003
+++ qhull-2003.1/src/Makefile	Thu Apr 10 18:54:48 2008
@@ -32,6 +32,11 @@
 #       FILES -- all other files
 #       OBJS -- specifies the object files of libqhull.a
 #
+SRCDIR = @SRCDIR@/src
+
+vpath %.c $(SRCDIR)
+vpath %.h $(SRCDIR)
+
 BINDIR  = /usr/local/bin
 MANDIR  = /usr/local/man/man1
 
@@ -43,8 +48,8 @@
 
 #for Gnu's gcc compiler -O2 for optimization, -g for debugging, -Wall for check
 #
-CC     = gcc
-CCOPTS1 = -O2 -ansi 
+#CC     = gcc
+CCOPTS1 = $(GCC_ARCH_FLAGS) -O0
 
 # for Sun's cc compiler, -fast or O2 for optimization, -g for debugging, -Xc for ANSI
 #CC = cc
@@ -80,6 +85,17 @@
 		qdelau_f.htm qhalf.htm qvoronoi.htm qvoron_f.htm rbox.htm 
 
 all: rbox qconvex qdelaunay qhalf qvoronoi qhull
+
+lib: libqhull.a qhull.dll
+
+qhull.dll : libqhull.a
+	$(CC) -shared $(LDFLAGS) -o $@ \
+	-Wl,--export-all-symbols \
+	-Wl,--enable-runtime-pseudo-reloc \
+	-Wl,--out-implib=$(<:.a=.dll.a) \
+	-Wl,--whole-archive $< \
+	-Wl,--no-whole-archive
+	strip $(STRIP_FLAGS) $@
 
 unix.o:   qhull.h user.h mem.h
 qconvex.o:   qhull.h user.h mem.h
