diff -urN -x *.o -x *.dll -x *.a -x *.exe -x *.def glpk-4.17-orig/src/Makefile.in glpk-4.17/src/Makefile.in
--- glpk-4.17-orig/src/Makefile.in	Sat May 26 10:00:00 2007
+++ glpk-4.17/src/Makefile.in	Thu Jun 28 22:00:49 2007
@@ -187,7 +187,7 @@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 INCLUDES = -I../include
-lib_LTLIBRARIES = libglpk.la
+lib_LTLIBRARIES = libglpk.la glpk.dll
 libglpk_la_LDFLAGS = -version-info 2:0:2
 libglpk_la_SOURCES = \
 glpapi01.c \
@@ -322,6 +322,16 @@
 	done
 libglpk.la: $(libglpk_la_OBJECTS) $(libglpk_la_DEPENDENCIES) 
 	$(libglpk_la_LINK) -rpath $(libdir) $(libglpk_la_OBJECTS) $(libglpk_la_LIBADD) $(LIBS)
+
+	glpk.dll : libglpk.la
+	$(CC) -shared -o $@ \
+	  -Wl,--enable-auto-import \
+	  -Wl,--export-all-symbols \
+	  -Wl,--whole-archive .libs/libglpk.a \
+	  -Wl,--out-implib=libglpk.dll.a \
+	  -Wl,--output-def=libglpk.def \
+	  -Wl,--no-whole-archive $(LIBS)
+	strip $(STRIP_FLAGS) $@
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
