npp: local src changes (bugfixes)

diff -r 390c29f033a4 PowerEditor/src/MISC/Common/precompiledHeaders.h
--- a/PowerEditor/src/MISC/Common/precompiledHeaders.h	Sat Feb 20 17:05:20 2010 +0100
+++ b/PowerEditor/src/MISC/Common/precompiledHeaders.h	Sat Feb 20 17:07:13 2010 +0100
@@ -50,7 +50,7 @@
 #include <shlwapi.h>
 #include <uxtheme.h>
 #include <Oleacc.h>
-#include <dbghelp.h>
+// #include <dbghelp.h>
 #include <eh.h>
 
 #ifdef UNICODE
@@ -62,4 +62,4 @@
 #include "Window.h"
 #include "StaticDialog.h"
 
-#endif PRECOMPILEHEADER_H
+#endif /* PRECOMPILEHEADER_H */
diff -r 390c29f033a4 PowerEditor/src/Notepad_plus.cpp
--- a/PowerEditor/src/Notepad_plus.cpp	Sat Feb 20 17:05:20 2010 +0100
+++ b/PowerEditor/src/Notepad_plus.cpp	Sat Feb 20 17:07:13 2010 +0100
@@ -2087,8 +2087,8 @@
 
 int Notepad_plus::doReloadOrNot(const TCHAR *fn, bool dirty) 
 {
-	TCHAR* pattern = TEXT("%s\r\rThis file has been modified by another program.\rDo you want to reload it%s?");
-	TCHAR* lose_info_str = dirty ? TEXT(" and lose the changes made in Notepad++") : TEXT("");
+	const TCHAR* pattern = TEXT("%s\r\rThis file has been modified by another program.\rDo you want to reload it%s?");
+	const TCHAR* lose_info_str = dirty ? TEXT(" and lose the changes made in Notepad++") : TEXT("");
 	TCHAR phrase[512];
 	wsprintf(phrase, pattern, fn, lose_info_str);
 	int icon = dirty ? MB_ICONEXCLAMATION : MB_ICONQUESTION;
diff -r 390c29f033a4 PowerEditor/src/Notepad_plus.h
--- a/PowerEditor/src/Notepad_plus.h	Sat Feb 20 17:05:20 2010 +0100
+++ b/PowerEditor/src/Notepad_plus.h	Sat Feb 20 17:07:13 2010 +0100
@@ -110,6 +110,8 @@
 #include "lesDlgs.h"
 #endif //SIZE_DLG_H
 
+#include "shortcutmapper.h"
+
 #define MENU 0x01
 #define TOOLBAR 0x02
 
@@ -183,7 +185,7 @@
 	inline void killAllChildren();
 	virtual inline void destroy();
 
-    static const TCHAR * Notepad_plus::getClassName() {
+    static const TCHAR * getClassName() {
 		return _className;
 	};
 	
diff -r 390c29f033a4 PowerEditor/src/Notepad_plus.rc
--- a/PowerEditor/src/Notepad_plus.rc	Sat Feb 20 17:05:20 2010 +0100
+++ b/PowerEditor/src/Notepad_plus.rc	Sat Feb 20 17:07:13 2010 +0100
@@ -798,4 +798,4 @@
 END
 
 // xp style
-1 RT_MANIFEST "notepad++.exe.manifest"
+//1 RT_MANIFEST "notepad++.exe.manifest"
diff -r 390c29f033a4 PowerEditor/src/Parameters.h
--- a/PowerEditor/src/Parameters.h	Sat Feb 20 17:05:20 2010 +0100
+++ b/PowerEditor/src/Parameters.h	Sat Feb 20 17:07:13 2010 +0100
@@ -1078,7 +1078,7 @@
         _list.push_back(pair<generic_string, bool>(fn, isInBL));
     };
 private :
-    vector<pair<generic_string, bool>>_list;
+    vector< pair<generic_string, bool> >_list;
 };
 
 const int NB_LANG = 80;
diff -r 390c29f033a4 PowerEditor/src/ScitillaComponent/Buffer.h
--- a/PowerEditor/src/ScitillaComponent/Buffer.h	Sat Feb 20 17:05:20 2010 +0100
+++ b/PowerEditor/src/ScitillaComponent/Buffer.h	Sat Feb 20 17:07:13 2010 +0100
@@ -59,6 +59,8 @@
 //const int userLangNameMax = 16;
 const TCHAR UNTITLED_STR[] = TEXT("new ");
 
+class Notepad_plus;
+
 //File manager class maintains all buffers
 class FileManager {
 public:
diff -r 390c29f033a4 PowerEditor/src/ScitillaComponent/FindReplaceDlg.rc
--- a/PowerEditor/src/ScitillaComponent/FindReplaceDlg.rc	Sat Feb 20 17:05:20 2010 +0100
+++ b/PowerEditor/src/ScitillaComponent/FindReplaceDlg.rc	Sat Feb 20 17:07:13 2010 +0100
@@ -20,6 +20,7 @@
 #ifndef FIND_REPLACE_DLG_RC
 #define FIND_REPLACE_DLG_RC
 #include <windows.h>
+#include <commctrl.h>
 #include "FindReplaceDlg_rc.h"
 
 IDD_FIND_REPLACE_DLG DIALOGEX 36, 44, 317, 182
diff -r 390c29f033a4 PowerEditor/src/ScitillaComponent/Printer.h
--- a/PowerEditor/src/ScitillaComponent/Printer.h	Sat Feb 20 17:05:20 2010 +0100
+++ b/PowerEditor/src/ScitillaComponent/Printer.h	Sat Feb 20 17:07:13 2010 +0100
@@ -36,7 +36,7 @@
 public :
 	Printer(){};
 	void init(HINSTANCE hInst, HWND hwnd, ScintillaEditView *pSEView, bool showDialog, int startPos, int endPos);
-	size_t Printer::doPrint() {
+	size_t doPrint() {
 		if (!::PrintDlg(&_pdlg))
 				return 0;
 
diff -r 390c29f033a4 PowerEditor/src/ScitillaComponent/ScintillaEditView.cpp
--- a/PowerEditor/src/ScitillaComponent/ScintillaEditView.cpp	Sat Feb 20 17:05:20 2010 +0100
+++ b/PowerEditor/src/ScitillaComponent/ScintillaEditView.cpp	Sat Feb 20 17:07:13 2010 +0100
@@ -2173,8 +2173,11 @@
 
     range.first = execute(SCI_LINEFROMPOSITION, start);
     range.second = execute(SCI_LINEFROMPOSITION, end);
-    if (range.first > range.second)
-        range.swap(range);
+    if (range.first > range.second) {
+	int t = range.first;
+	range.first = range.second;
+	range.second = t;
+    }
     return range;
 }
 
@@ -2267,9 +2270,9 @@
              for (int j = 0 ; j < nbChar ; j++)
              {
                  if (Case == UPPERCASE)
-                     destStr[j] = (wchar_t)::CharUpperW((LPWSTR)destStr[j]);
+                     destStr[j] = (wchar_t)*::CharUpperW((LPWSTR)destStr[j]);
                  else
-                     destStr[j] = (wchar_t)::CharLowerW((LPWSTR)destStr[j]);
+                     destStr[j] = (wchar_t)*::CharLowerW((LPWSTR)destStr[j]);
              }
              ::WideCharToMultiByte(codepage, 0, destStr, len, srcStr, len, NULL, NULL);
   
@@ -2307,9 +2310,9 @@
 		for (int i = 0 ; i < nbChar ; i++)
 		{
 			if (Case == UPPERCASE)
-				selectedStrW[i] = (WCHAR)::CharUpperW((LPWSTR)selectedStrW[i]);
+				selectedStrW[i] = (WCHAR)*::CharUpperW((LPWSTR)selectedStrW[i]);
 			else
-				selectedStrW[i] = (WCHAR)::CharLowerW((LPWSTR)selectedStrW[i]);
+				selectedStrW[i] = (WCHAR)*::CharLowerW((LPWSTR)selectedStrW[i]);
 		}
 		::WideCharToMultiByte(codepage, 0, selectedStrW, strWSize, selectedStr, strSize, NULL, NULL);
 
diff -r 390c29f033a4 PowerEditor/src/ScitillaComponent/ScintillaEditView.h
--- a/PowerEditor/src/ScitillaComponent/ScintillaEditView.h	Sat Feb 20 17:05:20 2010 +0100
+++ b/PowerEditor/src/ScitillaComponent/ScintillaEditView.h	Sat Feb 20 17:07:13 2010 +0100
@@ -160,7 +160,7 @@
 // SortClass for vector<ColumnModeInfo>
 // sort in _order : increased order
 struct SortInSelectOrder {
-	bool operator() (ColumnModeInfo & l, ColumnModeInfo & r) {
+	bool operator() (const ColumnModeInfo & l, const ColumnModeInfo & r) {
 		return (l._order < r._order);
 	}
 };
@@ -169,7 +169,7 @@
 // SortClass for vector<ColumnModeInfo>
 // sort in _selLpos : increased order
 struct SortInPositionOrder {
-	bool operator() (ColumnModeInfo & l, ColumnModeInfo & r) {
+	bool operator() (const ColumnModeInfo & l, const ColumnModeInfo & r) {
 		return (l._selLpos < r._selLpos);
 	}
 };
@@ -532,7 +532,7 @@
 		execute(SCI_INDICATORCLEARRANGE, docStart, docEnd-docStart);
 	};
 
-	static LanguageName ScintillaEditView::langNames[L_EXTERNAL+1];
+	static LanguageName langNames[L_EXTERNAL+1];
 
 	void bufferUpdated(Buffer * buffer, int mask);
 	BufferID getCurrentBufferID() { return _currentBufferID; };
diff -r 390c29f033a4 PowerEditor/src/WinControls/TabBar/ControlsTab.h
--- a/PowerEditor/src/WinControls/TabBar/ControlsTab.h	Sat Feb 20 17:05:20 2010 +0100
+++ b/PowerEditor/src/WinControls/TabBar/ControlsTab.h	Sat Feb 20 17:07:13 2010 +0100
@@ -44,7 +44,7 @@
 		_isVertical = isVertical;
 		TabBar::init(hInst, hwnd, false, isTraditional, isMultiLine);
 	};
-	void ControlsTab::createTabs(WindowVector & winVector);
+	void createTabs(WindowVector & winVector);
 
 	void destroy() {
 		TabBar::destroy();
diff -r 390c29f033a4 PowerEditor/src/WinControls/shortcut/shortcut.h
--- a/PowerEditor/src/WinControls/shortcut/shortcut.h	Sat Feb 20 17:05:20 2010 +0100
+++ b/PowerEditor/src/WinControls/shortcut/shortcut.h	Sat Feb 20 17:07:13 2010 +0100
@@ -203,7 +203,7 @@
 	};
 
 	KeyCombo getKeyComboByIndex(int index) const;
-	void ScintillaKeyMap::setKeyComboByIndex(int index, KeyCombo combo);
+	void setKeyComboByIndex(int index, KeyCombo combo);
 	void removeKeyComboByIndex(int index);
 	void clearDups() {
 		if (size > 1)
