diff -r 306a0b25c3fb PowerEditor/src/Notepad_plus.h
--- a/PowerEditor/src/Notepad_plus.h	Mon Feb 09 10:00:30 2009 +0100
+++ b/PowerEditor/src/Notepad_plus.h	Mon Feb 09 10:06:38 2009 +0100
@@ -49,6 +49,7 @@
 #include "AutoCompletion.h"
 #include "Buffer.h"
 #include "SmartHighlighter.h"
+#include "shortcutmapper.h"
 
 #define MENU 0x01
 #define TOOLBAR 0x02
@@ -122,7 +123,7 @@
 	inline void killAllChildren();
 	virtual inline void destroy();
 
-    static const TCHAR * Notepad_plus::getClassName() {
+    static const TCHAR * getClassName() {
 		return _className;
 	};
 	
@@ -426,15 +427,15 @@
 //END: Document management
 
 	int doSaveOrNot(const TCHAR *fn) {
-		TCHAR pattern[64] = TEXT("Save file \"%s\" ?");
+		const TCHAR* pattern = TEXT("Save file \"%s\" ?");
 		TCHAR phrase[512];
 		wsprintf(phrase, pattern, fn);
 		return doActionOrNot(TEXT("Save"), phrase, MB_YESNOCANCEL | MB_ICONQUESTION | MB_APPLMODAL);
 	};
 
 	int doReloadOrNot(const TCHAR *fn, bool dirty) {
-		TCHAR* pattern = TEXT("%s\r\rThis file has been modified by another program.\rDo you want to reload it%s?");
-		TCHAR* lose_info_str = dirty ? TEXT(" and lose the changes made in Notepad++") : TEXT("");
+		const TCHAR* pattern = TEXT("%s\r\rThis file has been modified by another program.\rDo you want to reload it%s?");
+		const TCHAR* lose_info_str = dirty ? TEXT(" and lose the changes made in Notepad++") : TEXT("");
 		TCHAR phrase[512];
 		wsprintf(phrase, pattern, fn, lose_info_str);
 		int icon = dirty ? MB_ICONEXCLAMATION : MB_ICONQUESTION;
@@ -442,14 +443,14 @@
 	};
 
 	int doCloseOrNot(const TCHAR *fn) {
-		TCHAR pattern[128] = TEXT("The file \"%s\" doesn't exist anymore.\rKeep this file in editor?");
+		const TCHAR* pattern = TEXT("The file \"%s\" doesn't exist anymore.\rKeep this file in editor?");
 		TCHAR phrase[512];
 		wsprintf(phrase, pattern, fn);
 		return doActionOrNot(TEXT("Keep non existing file"), phrase, MB_YESNO | MB_ICONQUESTION | MB_APPLMODAL);
 	};
 	
 	int doDeleteOrNot(const TCHAR *fn) {
-		TCHAR pattern[128] = TEXT("The file \"%s\"\rwill be deleted from your disk and this document will be closed.\rContinue?");
+		const TCHAR* pattern = TEXT("The file \"%s\"\rwill be deleted from your disk and this document will be closed.\rContinue?");
 		TCHAR phrase[512];
 		wsprintf(phrase, pattern, fn);
 		return doActionOrNot(TEXT("Delete file"), phrase, MB_YESNO | MB_ICONQUESTION | MB_APPLMODAL);
diff -r 306a0b25c3fb PowerEditor/src/Notepad_plus.rc
--- a/PowerEditor/src/Notepad_plus.rc	Mon Feb 09 10:00:30 2009 +0100
+++ b/PowerEditor/src/Notepad_plus.rc	Mon Feb 09 10:06:38 2009 +0100
@@ -488,4 +488,4 @@
 END
 
 // xp style
-1 RT_MANIFEST "notepad++.exe.manifest"
+// 1 RT_MANIFEST "notepad++.exe.manifest"
diff -r 306a0b25c3fb PowerEditor/src/ScitillaComponent/Buffer.h
--- a/PowerEditor/src/ScitillaComponent/Buffer.h	Mon Feb 09 10:00:30 2009 +0100
+++ b/PowerEditor/src/ScitillaComponent/Buffer.h	Mon Feb 09 10:06:38 2009 +0100
@@ -58,6 +58,7 @@
 
 //File manager class maintains all buffers
 class Buffer;
+class Notepad_plus;
 class FileManager {
 public:
 	void init(Notepad_plus * pNotepadPlus, ScintillaEditView * pscratchTilla);
diff -r 306a0b25c3fb PowerEditor/src/ScitillaComponent/FindReplaceDlg.rc
--- a/PowerEditor/src/ScitillaComponent/FindReplaceDlg.rc	Mon Feb 09 10:00:30 2009 +0100
+++ b/PowerEditor/src/ScitillaComponent/FindReplaceDlg.rc	Mon Feb 09 10:06:38 2009 +0100
@@ -20,6 +20,7 @@
 #ifndef FIND_REPLACE_DLG_RC
 #define FIND_REPLACE_DLG_RC
 #include <windows.h>
+#include <commctrl.h>
 #include "FindReplaceDlg_rc.h"
 
 IDD_FIND_REPLACE_DLG DIALOGEX 36, 44, 317, 182
diff -r 306a0b25c3fb PowerEditor/src/ScitillaComponent/Printer.h
--- a/PowerEditor/src/ScitillaComponent/Printer.h	Mon Feb 09 10:00:30 2009 +0100
+++ b/PowerEditor/src/ScitillaComponent/Printer.h	Mon Feb 09 10:06:38 2009 +0100
@@ -36,7 +36,7 @@
 public :
 	Printer(){};
 	void init(HINSTANCE hInst, HWND hwnd, ScintillaEditView *pSEView, bool showDialog, int startPos, int endPos);
-	size_t Printer::doPrint() {
+	size_t doPrint() {
 		if (!::PrintDlg(&_pdlg))
 				return 0;
 
diff -r 306a0b25c3fb PowerEditor/src/ScitillaComponent/ScintillaEditView.cpp
--- a/PowerEditor/src/ScitillaComponent/ScintillaEditView.cpp	Mon Feb 09 10:00:30 2009 +0100
+++ b/PowerEditor/src/ScitillaComponent/ScintillaEditView.cpp	Mon Feb 09 10:06:38 2009 +0100
@@ -1888,9 +1888,9 @@
 			for (int j = 0 ; j < nbChar ; j++)
 			{
 				if (Case == UPPERCASE)
-					destStr[j] = (wchar_t)::CharUpperW((LPWSTR)destStr[j]);
+					destStr[j] = (wchar_t)*::CharUpperW((LPWSTR)destStr[j]);
 				else
-					destStr[j] = (wchar_t)::CharLowerW((LPWSTR)destStr[j]);
+					destStr[j] = (wchar_t)*::CharLowerW((LPWSTR)destStr[j]);
 			}
 			::WideCharToMultiByte(codepage, 0, destStr, len, srcStr, len, NULL, NULL);
 
@@ -1924,9 +1924,9 @@
 		for (int i = 0 ; i < nbChar ; i++)
 		{
 			if (Case == UPPERCASE)
-				selectedStrW[i] = (WCHAR)::CharUpperW((LPWSTR)selectedStrW[i]);
+				selectedStrW[i] = (WCHAR)*::CharUpperW((LPWSTR)selectedStrW[i]);
 			else
-				selectedStrW[i] = (WCHAR)::CharLowerW((LPWSTR)selectedStrW[i]);
+				selectedStrW[i] = (WCHAR)*::CharLowerW((LPWSTR)selectedStrW[i]);
 		}
 		::WideCharToMultiByte(codepage, 0, selectedStrW, strWSize, selectedStr, strSize, NULL, NULL);
 
diff -r 306a0b25c3fb PowerEditor/src/ScitillaComponent/ScintillaEditView.h
--- a/PowerEditor/src/ScitillaComponent/ScintillaEditView.h	Mon Feb 09 10:00:30 2009 +0100
+++ b/PowerEditor/src/ScitillaComponent/ScintillaEditView.h	Mon Feb 09 10:06:38 2009 +0100
@@ -456,7 +456,7 @@
 		execute(SCI_INDICATORCLEARRANGE, docStart, docEnd-docStart);
 	};
 
-	static LanguageName ScintillaEditView::langNames[L_EXTERNAL+1];
+	static LanguageName langNames[L_EXTERNAL+1];
 
 	void bufferUpdated(Buffer * buffer, int mask);
 	BufferID getCurrentBufferID() { return _currentBufferID; };
diff -r 306a0b25c3fb PowerEditor/src/WinControls/TabBar/ControlsTab.h
--- a/PowerEditor/src/WinControls/TabBar/ControlsTab.h	Mon Feb 09 10:00:30 2009 +0100
+++ b/PowerEditor/src/WinControls/TabBar/ControlsTab.h	Mon Feb 09 10:06:38 2009 +0100
@@ -52,7 +52,7 @@
 		//TabBar::init(hInst, hwnd, false, true);
 		TabBar::init(hInst, hwnd, false, isTraditional, isMultiLine);
 	};
-	void ControlsTab::createTabs(WindowVector & winVector);
+	void createTabs(WindowVector & winVector);
 
 	void destroy() {
 		TabBar::destroy();
diff -r 306a0b25c3fb PowerEditor/src/WinControls/shortcut/shortcut.h
--- a/PowerEditor/src/WinControls/shortcut/shortcut.h	Mon Feb 09 10:00:30 2009 +0100
+++ b/PowerEditor/src/WinControls/shortcut/shortcut.h	Mon Feb 09 10:06:38 2009 +0100
@@ -204,7 +204,7 @@
 	};
 
 	KeyCombo getKeyComboByIndex(int index) const;
-	void ScintillaKeyMap::setKeyComboByIndex(int index, KeyCombo combo);
+	void setKeyComboByIndex(int index, KeyCombo combo);
 	void removeKeyComboByIndex(int index);
 	void clearDups() {
 		if (size > 1)
