diff -r 5c33c42df98f PowerEditor/src/Parameters.cpp
--- a/PowerEditor/src/Parameters.cpp	Mon Feb 09 11:10:59 2009 +0100
+++ b/PowerEditor/src/Parameters.cpp	Mon Feb 09 11:12:11 2009 +0100
@@ -331,7 +331,7 @@
 	//
 };
 #ifdef UNICODE
-#include "localizationString.h"
+#include "mingw-localizationString.h"
 
 LocalizationSwicher::LocalizationSwicher() 
 {
diff -r 5c33c42df98f PowerEditor/src/mingw-localizationString.h
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/PowerEditor/src/mingw-localizationString.h	Mon Feb 09 11:12:11 2009 +0100
@@ -0,0 +1,23 @@
+//this file is part of notepad++
+//Copyright (C)2003 Don HO <donho@altern.org>
+//
+//This program is free software; you can redistribute it and/or
+//modify it under the terms of the GNU General Public License
+//as published by the Free Software Foundation; either
+//version 2 of the License, or (at your option) any later version.
+//
+//This program is distributed in the hope that it will be useful,
+//but WITHOUT ANY WARRANTY; without even the implied warranty of
+//MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//GNU General Public License for more details.
+//
+//You should have received a copy of the GNU General Public License
+//along with this program; if not, write to the Free Software
+//Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+
+#ifndef LOCALIZATIONSTRING_H
+#define LOCALIZATIONSTRING_H
+LocalizationSwicher::LocalizationDefinition localizationDefs[] = {
+	{TEXT("English"), TEXT("english.xml")},
+};
+#endif LOCALIZATIONSTRING_H
