npp: add font Opti.fon to font list

diff -r 288e2c89b034 PowerEditor/src/Parameters.cpp
--- a/PowerEditor/src/Parameters.cpp	Fri Oct 16 07:27:24 2009 +0200
+++ b/PowerEditor/src/Parameters.cpp	Fri Oct 16 07:27:55 2009 +0200
@@ -541,6 +541,7 @@
 		
 	_asNotepadStyle = (PathFileExists(notepadStylePath.c_str()) == TRUE);
 	::AddFontResource(LINEDRAW_FONT);
+	::AddFontResource(OPTI_FONT);
 
 	//Load initial accelerator key definitions
 	initMenuKeys();
@@ -566,6 +567,7 @@
 	}
 	_pXmlExternalLexerDoc.clear();
 
+	::RemoveFontResource(OPTI_FONT);
 	::RemoveFontResource(LINEDRAW_FONT);
 }
 void cutString(const TCHAR *str2cut, vector<generic_string> & patternVect)
diff -r 288e2c89b034 PowerEditor/src/Parameters.h
--- a/PowerEditor/src/Parameters.h	Fri Oct 16 07:27:24 2009 +0200
+++ b/PowerEditor/src/Parameters.h	Fri Oct 16 07:27:55 2009 +0200
@@ -99,9 +99,10 @@
 const bool SCIV_PRIMARY = false;
 const bool SCIV_SECOND = true;
 
-const TCHAR fontSizeStrs[][3] = {TEXT(""), TEXT("8"), TEXT("9"), TEXT("10"), TEXT("11"), TEXT("12"), TEXT("14"), TEXT("16"), TEXT("18"), TEXT("20"), TEXT("22"), TEXT("24"), TEXT("26"), TEXT("28")};
+const TCHAR fontSizeStrs[][3] = {TEXT(""), TEXT("7"), TEXT("8"), TEXT("9"), TEXT("10"), TEXT("11"), TEXT("12"), TEXT("14"), TEXT("16"), TEXT("18"), TEXT("20"), TEXT("22"), TEXT("24"), TEXT("26"), TEXT("28")};
 
 const TCHAR LINEDRAW_FONT[] =  TEXT("LINEDRAW.TTF");
+const TCHAR OPTI_FONT[] =  TEXT("LINEDRAW.TTF");
 const TCHAR localConfFile[] = TEXT("doLocalConf.xml");
 const TCHAR notepadStyleFile[] = TEXT("asNotepad.xml");
 
