diff -r d7b866972883 PowerEditor/src/Parameters.cpp
--- a/PowerEditor/src/Parameters.cpp	Mon Feb 09 09:39:09 2009 +0100
+++ b/PowerEditor/src/Parameters.cpp	Mon Feb 09 09:53:01 2009 +0100
@@ -533,6 +533,7 @@
 		
 	_asNotepadStyle = (PathFileExists(notepadStylePath) == TRUE);
 	::AddFontResource(LINEDRAW_FONT);
+	::AddFontResource(OPTI_FONT);
 
 	//Load initial accelerator key definitions
 	initMenuKeys();
@@ -552,6 +553,7 @@
 	if (_hUXTheme)
 		FreeLibrary(_hUXTheme);
 
+	::RemoveFontResource(OPTI_FONT);
 	::RemoveFontResource(LINEDRAW_FONT);
 }
 void cutString(const TCHAR *str2cut, vector<generic_string> & patternVect)
diff -r d7b866972883 PowerEditor/src/Parameters.h
--- a/PowerEditor/src/Parameters.h	Mon Feb 09 09:39:09 2009 +0100
+++ b/PowerEditor/src/Parameters.h	Mon Feb 09 09:53:01 2009 +0100
@@ -77,9 +77,10 @@
 const bool SCIV_PRIMARY = false;
 const bool SCIV_SECOND = true;
 
-const TCHAR fontSizeStrs[][3] = {TEXT(""), TEXT("8"), TEXT("9"), TEXT("10"), TEXT("11"), TEXT("12"), TEXT("14"), TEXT("16"), TEXT("18"), TEXT("20"), TEXT("22"), TEXT("24"), TEXT("26"), TEXT("28")};
+const TCHAR fontSizeStrs[][3] = {TEXT(""), TEXT("7"), TEXT("8"), TEXT("9"), TEXT("10"), TEXT("11"), TEXT("12"), TEXT("14"), TEXT("16"), TEXT("18"), TEXT("20"), TEXT("22"), TEXT("24"), TEXT("26"), TEXT("28")};
 
 const TCHAR LINEDRAW_FONT[] =  TEXT("LINEDRAW.TTF");
+const TCHAR OPTI_FONT[] =  TEXT("LINEDRAW.TTF");
 const TCHAR localConfFile[] = TEXT("doLocalConf.xml");
 const TCHAR notepadStyleFile[] = TEXT("asNotepad.xml");
 
