fix wrong definition of lmProgress in LexProgress.cxx

diff -r 4b4287ea830e scintilla/src/LexProgress.cxx
--- a/scintilla/src/LexProgress.cxx	Mon Feb 23 10:23:46 2009 +0100
+++ b/scintilla/src/LexProgress.cxx	Mon Feb 23 14:10:28 2009 +0100
@@ -268,4 +268,4 @@
             0,
         };
 
-LexerModule lmProgress(SCLEX_PS, Colourise4glDoc, "progress", Fold4glDoc, FglWordLists);
+LexerModule lmProgress(SCLEX_PROGRESS, Colourise4glDoc, "progress", Fold4glDoc, FglWordLists);
