scintilla: local bugfixes

diff -r a9cd7333b264 scintilla/src/LexSearchResult.cxx
--- a/scintilla/src/LexSearchResult.cxx	Tue Jul 07 11:57:32 2009 +0200
+++ b/scintilla/src/LexSearchResult.cxx	Tue Jul 07 13:11:20 2009 +0200
@@ -35,7 +35,7 @@
 #include "SciLexer.h"
 
 // The following definitions are a copy of the ones in FindReplaceDlg.h
-static enum { searchHeaderLevel = SC_FOLDLEVELBASE + 1, fileHeaderLevel, resultLevel };
+enum { searchHeaderLevel = SC_FOLDLEVELBASE + 1, fileHeaderLevel, resultLevel };
 
 
 static inline bool AtEOL(Accessor &styler, unsigned int i) {
