diff -r 168e89419413 scintilla/src/LexSearchResult.cxx
--- a/scintilla/src/LexSearchResult.cxx	Mon Feb 09 10:43:18 2009 +0100
+++ b/scintilla/src/LexSearchResult.cxx	Mon Feb 09 10:46:02 2009 +0100
@@ -35,7 +35,7 @@
 #include "SciLexer.h"
 
 // The following definitions are a copy of the ones in FindReplaceDlg.h
-static enum { searchHeaderLevel = SC_FOLDLEVELBASE + 1, fileHeaderLevel, resultLevel };
+enum { searchHeaderLevel = SC_FOLDLEVELBASE + 1, fileHeaderLevel, resultLevel };
 
 
 static inline bool AtEOL(Accessor &styler, unsigned int i) {
