diff -u -u _f2c/src/makefile f2c/src/makefile
--- ./f2c/src/makefile   2000-10-06 23:19:41.000000000 +0200
+++ ./f2c/src/makefile   2007-06-11 19:26:19.000000000 +0200
@@ -1,10 +1,10 @@
 #      Makefile for f2c, a Fortran 77 to C converter
 
 .SUFFIXES: .c .o
-CC = cc
-CFLAGS = -O
+: ${CC = cc}
+: ${CFLAGS = -O}
 SHELL = /bin/sh
-YACC = yacc
+: ${YACC = yacc}
 YFLAGS =
 
 .c.o:
--- ./f2c/libf2c/makefile	2000-10-06 23:19:47.000000000 +0200
+++ ./f2c/libf2c/makefile	2007-06-11 19:30:29.000000000 +0200
@@ -10,9 +10,9 @@
 #	-DNON_UNIX_STDIO
 # to the CLFAGS = line below.
 .SUFFIXES: .c .o
-CC = cc
+: ${CC = cc}
 SHELL = /bin/sh
-CFLAGS = -O
+: ${CFLAGS = -O}
 
 # compile, then strip unnecessary symbols
 .c.o:
@@ -70,10 +70,10 @@
 ### If your system lacks ranlib, you don't need it; see README.
 
 f77vers.o: f77vers.c
-	$(CC) -c f77vers.c
+	$(CC) $(CFLAGS) -c f77vers.c
 
 i77vers.o: i77vers.c
-	$(CC) -c i77vers.c
+	$(CC) $(CFLAGS) -c i77vers.c
 
 # To get an "f2c.h" for use with "f2c -C++", first "make hadd"
 hadd: f2c.h0 f2ch.add
