diff -urN -x '*.exe' -x '*.dll' -x '*.o' -x '*.a' -x '*.bak' libpng-1.2.32-orig/scripts/makefile.mingw libpng-1.2.32/scripts/makefile.mingw
--- libpng-1.2.32-orig/scripts/makefile.mingw	2008-09-18 11:48:05 +0200
+++ libpng-1.2.32/scripts/makefile.mingw	2008-11-09 19:30:38 +0100
@@ -10,6 +10,7 @@
 # Copyright (C) 1996, 1997 Andreas Dilger
 # For conditions of distribution and use, see copyright notice in png.h
 
+SRCDIR=@SRCDIR@
 
 # This makefile intends to support building outside the src directory
 # if desired. When invoking it, specify an argument to SRCDIR on the
@@ -33,7 +34,7 @@
 
 # If you're using a cross-compiler, add the appropriate prefix (e.g.,
 # "i386-mingw32msvc-") to the following three commands:
-CC=gcc
+#CC=gcc
 AR=ar
 RANLIB=ranlib
 
@@ -67,9 +68,9 @@
 #	-W -Wall -O $(ALIGN) -funroll-loops \
 #	-fomit-frame-pointer)  # $(WARNMORE) -g -DPNG_DEBUG=5
 ### Normal (non-asm) compilation
-CFLAGS= $(strip $(MINGW_CCFLAGS) $(addprefix -I,$(ZLIBINC)) \
-        -W -Wall -O3 $(ALIGN) -funroll-loops -DPNG_NO_MMX_CODE \
-	-fomit-frame-pointer) # $(WARNMORE) -g -DPNG_DEBUG=5
+CFLAGS += \
+        -Wall -O3 $(ALIGN) -funroll-loops -DPNG_NO_MMX_CODE \
+	-fomit-frame-pointer # $(WARNMORE) -g -DPNG_DEBUG=5
 
 LIBNAME = libpng12
 PNGMAJ = 0
@@ -77,16 +78,17 @@
 PNGMIN = 1.2.32
 PNGVER = $(PNGMAJ).$(PNGMIN)
 
-SHAREDLIB=libpng$(MINGDLL).dll
+SHAREDLIB=png$(MINGDLL).dll
 STATLIB=libpng.a
 IMPLIB=libpng.dll.a
 SHAREDDEF=libpng.def
 LIBS=$(SHAREDLIB) $(STATLIB)
 EXE=.exe
 
-LDFLAGS=$(strip -L. $(MINGW_LDFLAGS) -lpng $(addprefix -L,$(ZLIBLIB)) -lz)
-LDSFLAGS=$(strip -shared -L.  $(MINGW_LDFLAGS))
-LDEXTRA=-Wl,--out-implib=$(IMPLIB) $(addprefix -L,$(ZLIBLIB)) -lz
+#LDFLAGS=$(strip -L. $(MINGW_LDFLAGS) -lpng $(addprefix -L,$(ZLIBLIB)) -lz)
+#LDSFLAGS=$(strip -shared -L.  $(MINGW_LDFLAGS))
+LDSFLAGS=$(LDFLAGS)
+LDEXTRA=-Wl,--out-implib=$(IMPLIB) -lz
 
 INCPATH=$(prefix)/include
 LIBPATH=$(exec_prefix)/lib
@@ -113,9 +115,8 @@
 
 %.o : %.c
 	$(CC) -c $(CFLAGS) -o $@ $<
-%.pic.o : CFLAGS += -DPNG_BUILD_DLL
 %.pic.o : %.c
-	$(CC) -c $(CFLAGS) -o $@ $<
+	$(CC) -c $(CFLAGS) -DPNG_BUILD_DLL -o $@ $<
 
 all: all-static all-shared libpng.pc libpng-config libpng.pc libpng-config
 
@@ -127,15 +128,15 @@
  $(addprefix $(D),$(INCPATH)'\n' $(LIBPATH)'\n' $(BINPATH)'\n' \
    $(MANPATH)'\n' $(MAN3PATH)'\n' $(MAN5PATH)'\n')'\n'
 
-libpng.pc: scripts/libpng.pc.in
+libpng.pc: $(SRCDIR)/scripts/libpng.pc.in
 	@echo -e Making pkg-config file for this libpng installation..'\n' \
            using PREFIX=\"$(prefix)\"'\n'
-	cat scripts/libpng.pc.in | sed -e s!@prefix@!$(prefix)! \
+	cat $< | sed -e s!@prefix@!$(prefix)! \
 	-e s!@exec_prefix@!$(exec_prefix)! \
 	-e s!@libdir@!$(LIBPATH)! \
 	-e s!@includedir@!$(INCPATH)! \
 	-e s!@includedir@!$(INCPATH)! \
-	-e s!-lpng12!-lpng12\ -lz\ -lm! > libpng.pc
+	-e s!-lpng12!-lpng\ -lz\ -lm! > libpng.pc
 
 libpng-config: scripts/libpng-config-head.in scripts/libpng-config-body.in
 	@echo -e Making $(LIBNAME) libpng-config file for this libpng \
@@ -162,13 +163,13 @@
 	sed -e 's/\([^;]*\);/;/' > $@
 
 $(SHAREDLIB): $(OBJSDLL) $(SHAREDDEF)
-	$(CC) $(LDSFLAGS) -o $@ $(OBJSDLL) -L. $(LDEXTRA)
+	$(CC) -shared $(LDSFLAGS) -o $@ $(OBJSDLL) $(SHAREDDEF) -L. $(LDEXTRA)
 
 pngtest$(EXE): pngtest.pic.o $(SHAREDLIB)
-	$(CC) $(CFLAGS) $< $(LDFLAGS) -o $@
+	$(CC) -WL,-Bdynamic $< $(LDFLAGS) -o $@ -L. -lpng -lz
 
 pngtest-stat$(EXE): pngtest.o $(STATLIB)
-	$(CC) -static $(CFLAGS) $< $(LDFLAGS) -o $@
+	$(CC) -Wl,-Bstatic $< $(LDFLAGS) -o $@ -L. -lpng -L$(prefix)/staticlib -lz
 
 test: test-static test-shared
 
