diff -urN -x '*.exe' -x '*.dll' -x '*.o' -x '*.a' -x '*.bak' -x '~' -x .hg gd-2.0.36RC1-orig/Makefile.in gd-2.0.36RC1/Makefile.in
--- gd-2.0.36RC1-orig/Makefile.in	2009-08-30 15:51:16 +0200
+++ gd-2.0.36RC1/Makefile.in	2009-08-30 15:51:16 +0200
@@ -351,7 +351,7 @@
 bin_SCRIPTS = bdftogd config/gdlib-config
 EXTRA_DIST = README-JPEG.TXT README.TXT configure.pl bdftogd demoin.png err.out index.html install-item makefile.sample readme.jpn entities.html entities.tcl
 include_HEADERS = gd.h gdfx.h gd_io.h gdcache.h gdfontg.h gdfontl.h gdfontmb.h gdfonts.h gdfontt.h entities.h
-lib_LTLIBRARIES = libgd.la
+lib_LTLIBRARIES = libgd.la gd.dll
 libgd_la_SOURCES = gd.c gdfx.c gd_security.c gd_gd.c gd_gd2.c gd_io.c gd_io_dp.c gd_gif_in.c gd_gif_out.c gd_io_file.c gd_io_ss.c gd_jpeg.c gd_png.c gd_ss.c gd_topal.c gd_wbmp.c gdcache.c gdfontg.c gdfontl.c gdfontmb.c gdfonts.c gdfontt.c gdft.c gdhelpers.c gdhelpers.h gdkanji.c gdtables.c gdxpm.c jisx0208.h wbmp.c wbmp.h
 libgd_la_LDFLAGS = -version-info 2:0:0 $(XTRA_LDFLAGS)
 libgd_la_LIBADD = $(LTLIBICONV)
@@ -440,6 +440,14 @@
 	done
 libgd.la: $(libgd_la_OBJECTS) $(libgd_la_DEPENDENCIES) 
 	$(libgd_la_LINK) -rpath $(libdir) $(libgd_la_OBJECTS) $(libgd_la_LIBADD) $(LIBS)
+ 
+gd.dll : libgd.la
+	$(CC) -shared $(LDFLAGS) -o $@ \
+	-Wl,--out-implib=libgd.dll.a \
+	-Wl,--output-def=$@.exp \
+	-Wl,--whole-archive ./.libs/libgd.a \
+	-Wl,--no-whole-archive $(LIBS)
+
 install-binPROGRAMS: $(bin_PROGRAMS)
 	@$(NORMAL_INSTALL)
 	test -z "$(bindir)" || $(MKDIR_P) "$(DESTDIR)$(bindir)"
diff -urN -x '*.exe' -x '*.dll' -x '*.o' -x '*.a' -x '*.bak' -x '~' -x .hg gd-2.0.36RC1-orig/gd.h gd-2.0.36RC1/gd.h
--- gd-2.0.36RC1-orig/gd.h	2009-08-30 15:51:16 +0200
+++ gd-2.0.36RC1/gd.h	2009-08-30 15:51:46 +0200
@@ -70,12 +70,20 @@
 
 /* 2.0.23: more Type 1 fonts */
 #ifndef DEFAULT_FONTPATH
+#ifdef WIN32
+#define DEFAULT_FONTPATH "C:\\WINDOWS\\FONTS;C:\\WINNT\\FONTS"
+#else
 /* default fontpath for unix systems  - whatever happened to standards ! */
 #define DEFAULT_FONTPATH "/usr/X11R6/lib/X11/fonts/TrueType:/usr/X11R6/lib/X11/fonts/truetype:/usr/X11R6/lib/X11/fonts/TTF:/usr/share/fonts/TrueType:/usr/share/fonts/truetype:/usr/openwin/lib/X11/fonts/TrueType:/usr/X11R6/lib/X11/fonts/Type1:/usr/lib/X11/fonts/Type1:/usr/openwin/lib/X11/fonts/Type1"
 #endif
+#endif
 #ifndef PATHSEPARATOR
+#ifdef WIN32
+#define PATHSEPARATOR ";"
+#else
 #define PATHSEPARATOR ":"
 #endif
+#endif
 
 /* gd.h: declarations file for the graphic-draw module.
  * Permission to use, copy, modify, and distribute this software and its
diff -urN -x '*.exe' -x '*.dll' -x '*.o' -x '*.a' -x '*.bak' -x '~' -x .hg -x '*.orig' gd-2.0.36RC1-orig/gd_jpeg.c gd-2.0.36RC1/gd_jpeg.c
--- gd-2.0.36RC1-orig/gd_jpeg.c	2009-09-27 17:30:49 +0200
+++ gd-2.0.36RC1/gd_jpeg.c	2009-09-27 17:21:38 +0200
@@ -42,6 +42,11 @@
    redefined incompatibly in jpeglib.h */
 #undef HAVE_STDDEF_H
 #undef HAVE_STDLIB_H
+
+#ifdef gdHACK_BOOLEAN
+#define HAVE_BOOLEAN
+#define boolean gdHACK_BOOLEAN
+#endif
 
 /* 1.8.1: remove dependency on jinclude.h */
 #include "jpeglib.h"
