diff -urN -x *.exe -x *.dll -x *.o -x *.a -x *.bak tiff-3.8.2-orig/libtiff/Makefile.in tiff-3.8.2/libtiff/Makefile.in
--- tiff-3.8.2-orig/libtiff/Makefile.in	Thu Mar 23 15:44:45 2006
+++ tiff-3.8.2/libtiff/Makefile.in	Tue Jan  1 16:39:17 2008
@@ -353,7 +353,7 @@
 libtiffincludedir = $(includedir)
 libtiffinclude_HEADERS = $(HDRS)
 noinst_HEADERS = $(EXTRA_HDRS)
-lib_LTLIBRARIES = libtiff.la $(am__append_2)
+lib_LTLIBRARIES = libtiff.la $(am__append_2) libtiff.dll libtiffxx.dll
 libtiff_la_SOURCES = $(SRCS)
 libtiff_la_LDFLAGS = -no-undefined -version-number \
 	$(LIBTIFF_VERSION_INFO) $(am__append_3)
@@ -473,6 +473,23 @@
 
 distclean-compile:
 	-rm -f *.tab.c
+
+libtiff.dll : libtiff.la
+	$(CC) -shared $(LDFLAGS) -o libtiff.dll \
+	-Wl,--export-all-symbols \
+	-Wl,--whole-archive ./.libs/libtiff.a \
+	-Wl,--out-implib=$(basename $@).dll.a \
+	-Wl,--no-whole-archive -lz -ljpeg
+	strip $(STRIP_FLAGS) $@
+
+libtiffxx.dll : libtiffxx.la
+	$(CXX) -shared $(LDFLAGS) -o libtiffxx.dll \
+	-Wl,--export-all-symbols \
+	-Wl,--whole-archive ./.libs/libtiffxx.a \
+	-Wl,--out-implib=$(basename $@).dll.a \
+	-Wl,--no-whole-archive libtiff.dll.a -lz -ljpeg
+	strip $(STRIP_FLAGS) $@
+
 
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/mkg3states.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/tif_aux.Plo@am__quote@
