diff -urN -x *.o -x *.dll -x *.a -x *.exe -x *.def -x *.out SuiteSparse-orig/AMD/Source/GNUmakefile SuiteSparse/AMD/Source/GNUmakefile
--- SuiteSparse-orig/AMD/Source/GNUmakefile	Tue May  2 13:12:27 2006
+++ SuiteSparse/AMD/Source/GNUmakefile	Sun Jun 10 20:14:37 2007
@@ -2,7 +2,7 @@
 # AMD Makefile for compiling on Unix systems (for GNU make only)
 #-------------------------------------------------------------------------------
 
-default: ../Lib/libamd.a
+default: ../Lib/libamd.a ../Lib/amd.dll
 
 include ../../UFconfig/UFconfig.mk
 
@@ -46,6 +46,15 @@
 ../Lib/libamd.a: amd_global.o $(AMDI) $(AMDL)
 	$(AR) ../Lib/libamd.a $^
 	- $(RANLIB) ../Lib/libamd.a
+
+../Lib/amd.dll : ../Lib/libamd.a
+	$(CC) -shared $(SHARED_FLAGS) -o $@ \
+		-Wl,--whole-archive $< \
+		-Wl,--out-implib,$(basename $<).dll.a \
+		-Wl,--output-def,$(basename $<).def \
+		-Wl,--no-whole-archive
+	$(STRIP) $(STRIP_FLAGS) $@
+
 
 #-------------------------------------------------------------------------------
 # compile the Fortran versions and the libamdf77.a library
diff -urN -x *.o -x *.dll -x *.a -x *.exe -x *.def -x *.out SuiteSparse-orig/CAMD/Source/GNUmakefile SuiteSparse/CAMD/Source/GNUmakefile
--- SuiteSparse-orig/CAMD/Source/GNUmakefile	Tue May  2 13:19:43 2006
+++ SuiteSparse/CAMD/Source/GNUmakefile	Sun Jun 10 20:14:37 2007
@@ -2,7 +2,7 @@
 # CAMD Makefile for compiling on Unix systems (for GNU make only)
 #-------------------------------------------------------------------------------
 
-default: ../Lib/libcamd.a
+default: ../Lib/libcamd.a ../Lib/camd.dll
 
 include ../../UFconfig/UFconfig.mk
 
@@ -46,6 +46,14 @@
 ../Lib/libcamd.a: camd_global.o $(CAMDI) $(CAMDL)
 	$(AR) ../Lib/libcamd.a $^
 	- $(RANLIB) ../Lib/libcamd.a
+
+../Lib/camd.dll : ../Lib/libcamd.a
+	$(CC) -shared $(SHARED_FLAGS) -o $@ \
+		-Wl,--whole-archive $< \
+		-Wl,--out-implib,$(basename $<).dll.a \
+		-Wl,--output-def,$(basename $<).def \
+		-Wl,--no-whole-archive
+	$(STRIP) $(STRIP_FLAGS) $@
 
 #-------------------------------------------------------------------------------
 # Remove all but the files in the original distribution
diff -urN -x *.o -x *.dll -x *.a -x *.exe -x *.def -x *.out SuiteSparse-orig/CCOLAMD/Makefile SuiteSparse/CCOLAMD/Makefile
--- SuiteSparse-orig/CCOLAMD/Makefile	Tue Jan 30 14:09:28 2007
+++ SuiteSparse/CCOLAMD/Makefile	Sun Jun 10 20:14:37 2007
@@ -1,5 +1,5 @@
 
-default: ccolamd_example ccolamd_l_example libccolamd.a
+default: ccolamd_example ccolamd_l_example libccolamd.a ccolamd.dll
 
 include ../UFconfig/UFconfig.mk
 
@@ -43,9 +43,17 @@
 	$(CC) $(CFLAGS) $(I) -c ccolamd.c -DDLONG -o ccolamd_l.o
 	$(AR) libccolamd.a ccolamd.o ccolamd_l.o ccolamd_global.o
 
+ccolamd.dll : libccolamd.a
+	$(CC) -shared $(SHARED_FLAGS) -o $@ \
+		-Wl,--whole-archive $< \
+		-Wl,--out-implib,$(basename $<).dll.a \
+		-Wl,--output-def,$(basename $<).def \
+		-Wl,--no-whole-archive
+	$(STRIP) $(STRIP_FLAGS) $@
+
 ccode: libccolamd.a
 
-library: libccolamd.a
+library: libccolamd.a ccolamd.dll
 
 clean:
 	- $(RM) $(CLEAN)
diff -urN -x *.o -x *.dll -x *.a -x *.exe -x *.def -x *.out SuiteSparse-orig/CHOLMOD/Lib/Makefile SuiteSparse/CHOLMOD/Lib/Makefile
--- SuiteSparse-orig/CHOLMOD/Lib/Makefile	Tue Oct 24 16:09:35 2006
+++ SuiteSparse/CHOLMOD/Lib/Makefile	Fri Jun 15 21:52:46 2007
@@ -10,9 +10,9 @@
 
 C = $(CC) $(CFLAGS) $(CHOLMOD_CONFIG)
 
-all: libcholmod.a
+all: libcholmod.a cholmod.dll
 
-library: libcholmod.a
+library: libcholmod.a cholmod.dll
 
 purge: distclean
 
@@ -112,6 +112,14 @@
 libcholmod.a: $(OBJ)
 	$(AR) libcholmod.a $(OBJ)
 	$(RANLIB) libcholmod.a
+
+cholmod.dll : libcholmod.a
+	$(CC) -shared $(SHARED_FLAGS) -o $@ -L../../AMD/Lib -L../../COLAMD \
+		-Wl,--whole-archive $< \
+		-Wl,--out-implib,$(basename $<).dll.a \
+		-Wl,--output-def,$(basename $<).def \
+		-Wl,--no-whole-archive -lamd -lcolamd $(BLAS) $(LAPACK)
+	$(STRIP) $(STRIP_FLAGS) $@
 
 $(OBJ): $(INC)
 
diff -urN -x *.o -x *.dll -x *.a -x *.exe -x *.def -x *.out SuiteSparse-orig/COLAMD/Makefile SuiteSparse/COLAMD/Makefile
--- SuiteSparse-orig/COLAMD/Makefile	Wed Dec 13 19:42:44 2006
+++ SuiteSparse/COLAMD/Makefile	Sun Jun 10 20:14:37 2007
@@ -1,5 +1,5 @@
 
-default: libcolamd.a colamd_example colamd_l_example
+default: libcolamd.a colamd_example colamd_l_example colamd.dll
 
 include ../UFconfig/UFconfig.mk
 
@@ -43,9 +43,17 @@
 	$(CC) $(CFLAGS) $(I) -c colamd.c -DDLONG -o colamd_l.o
 	$(AR) libcolamd.a colamd.o colamd_l.o colamd_global.o
 
+colamd.dll : libcolamd.a
+	$(CC) -shared $(SHARED_FLAGS) -o $@ \
+		-Wl,--whole-archive $< \
+		-Wl,--out-implib,$(basename $<).dll.a \
+		-Wl,--output-def,$(basename $<).def \
+		-Wl,--no-whole-archive 
+	$(STRIP) $(STRIP_FLAGS) $@
+
 ccode: libcolamd.a
 
-library: libcolamd.a
+library: libcolamd.a colamd.dll
 
 clean:
 	- $(RM) $(CLEAN)
diff -urN -x *.o -x *.dll -x *.a -x *.exe -x *.def -x *.out SuiteSparse-orig/CSparse/Source/Makefile SuiteSparse/CSparse/Source/Makefile
--- SuiteSparse-orig/CSparse/Source/Makefile	Wed Apr 12 16:15:34 2006
+++ SuiteSparse/CSparse/Source/Makefile	Sun Jun 10 20:14:37 2007
@@ -1,11 +1,13 @@
 # Modify the "-O" optimization option for best performance (-O3 on Linux):
-CC = cc
-CFLAGS = -O
+#CC = cc
+#CFLAGS = -O
+
+include ../../UFconfig/UFconfig.mk
 
 AR = ar cr
 RANLIB = ranlib
 
-all: libcsparse.a
+all: libcsparse.a csparse.dll
 
 CS = cs_add.o cs_amd.o cs_chol.o cs_cholsol.o cs_counts.o cs_cumsum.o \
 	cs_droptol.o cs_dropzeros.o cs_dupl.o cs_entry.o \
@@ -22,6 +24,14 @@
 libcsparse.a: $(CS)
 	$(AR) libcsparse.a $(CS)
 	$(RANLIB) libcsparse.a
+
+csparse.dll : libcsparse.a
+	$(CC) -shared $(SHARED_FLAGS) -o $@ \
+		-Wl,--whole-archive $< \
+		-Wl,--out-implib,$(basename $<).dll.a \
+		-Wl,--output-def,$(basename $<).def \
+		-Wl,--no-whole-archive
+	$(STRIP) $(STRIP_FLAGS) $@
 
 clean:
 	rm -f *.o
diff -urN -x *.o -x *.dll -x *.a -x *.exe -x *.def -x *.out SuiteSparse-orig/CXSparse/Source/Makefile SuiteSparse/CXSparse/Source/Makefile
--- SuiteSparse-orig/CXSparse/Source/Makefile	Tue May  2 14:06:19 2006
+++ SuiteSparse/CXSparse/Source/Makefile	Sun Jun 10 20:14:37 2007
@@ -1,12 +1,15 @@
 # Modify the "-O" optimization option for best performance (-O3 on Linux):
-CC = cc
-CFLAGS = -O
+#CC = cc
+#CFLAGS = -O
+
 I = -I../../UFconfig
 
 AR = ar cr
 RANLIB = ranlib
 
-all: libcxsparse.a
+include ../../UFconfig/UFconfig.mk
+
+all: libcxsparse.a cxsparse.dll
 
 CS_SOURCE = cs_add.c cs_amd.c cs_chol.c cs_cholsol.c cs_counts.c cs_cumsum.c \
 	cs_droptol.c cs_dropzeros.c cs_dupl.c cs_entry.c \
@@ -88,6 +91,14 @@
 libcxsparse.a: $(CS)
 	$(AR) libcxsparse.a $(CS)
 	$(RANLIB) libcxsparse.a
+
+cxsparse.dll : libcxsparse.a
+	$(CC) -shared $(SHARED_FLAGS) -o $@ \
+		-Wl,--whole-archive $< \
+		-Wl,--out-implib,$(basename $<).dll.a \
+		-Wl,--output-def,$(basename $<).def \
+		-Wl,--no-whole-archive
+	$(STRIP) $(STRIP_FLAGS) $@
 
 clean:
 	rm -f *.o
diff -urN -x *.o -x *.dll -x *.a -x *.exe -x *.def -x *.out SuiteSparse-orig/UFconfig/UFconfig.mk SuiteSparse/UFconfig/UFconfig.mk
--- SuiteSparse-orig/UFconfig/UFconfig.mk	Tue Dec 12 19:52:47 2006
+++ SuiteSparse/UFconfig/UFconfig.mk	Fri Jun 15 21:50:20 2007
@@ -31,8 +31,8 @@
 # C compiler and compiler flags:  These will normally not give you optimal
 # performance.  You should select the optimization parameters that are best
 # for your system.  On Linux, use "CFLAGS = -O3 -fexceptions" for example.
-CC = cc
-CFLAGS = -O
+CC = mingw32-gcc
+CFLAGS = -O3 -fexceptions
 
 # ranlib, and ar, for generating libraries
 RANLIB = ranlib
@@ -43,8 +43,8 @@
 MV = mv -f
 
 # Fortran compiler (not normally required)
-F77 = f77
-F77FLAGS = -O
+F77 = mingw32-f77
+F77FLAGS = -O3
 F77LIB =
 
 # C and Fortran libraries
@@ -53,6 +53,10 @@
 # For compiling MATLAB mexFunctions
 MEX = mex -O
 
+STRIP=strip
+
+SHARED_FLAGS=-Wl,--export-all-symbols 
+
 # Which version of MAKE you are using (default is "make")
 # MAKE = make
 # MAKE = gmake
@@ -73,8 +77,9 @@
 # These settings will probably not work, since there is no fixed convention for
 # naming the BLAS and LAPACK library (*.a or *.so) files.  Assume the Goto
 # BLAS are available.
-BLAS = -lgoto -lgfortran -lgfortranbegin
+BLAS = -lblas -lcblas
 LAPACK = -llapack
+BLASDIR = 
 
 # The BLAS might not contain xerbla, an error-handling routine for LAPACK and
 # the BLAS.  Also, the standard xerbla requires the Fortran I/O library, and
@@ -107,8 +112,8 @@
 
 # If you use CHOLMOD_CONFIG = -DNPARTITION then you must use the following
 # options:
-# METIS_PATH =
-# METIS =
+ METIS_PATH =
+ METIS =
 
 #------------------------------------------------------------------------------
 # UMFPACK configuration:
@@ -164,7 +169,7 @@
 # -DNSUNPERF	    for Solaris only.  If defined, do not use the Sun
 #			Performance Library
 
-CHOLMOD_CONFIG =
+CHOLMOD_CONFIG = -DNPARTITION
 
 #------------------------------------------------------------------------------
 # Linux
@@ -181,7 +186,7 @@
 # CFLAGS = -O3 -fexceptions \
    	-Wall -W -Werror -Wshadow -Wmissing-prototypes -Wstrict-prototypes \
     	-Wredundant-decls -Wnested-externs -Wdisabled-optimization -ansi
-CFLAGS = -O3 -fexceptions -D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE
+# CFLAGS = -O3 -fexceptions -D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE
 # CFLAGS = -O3
 
 # consider:
@@ -295,4 +300,4 @@
 # remove object files and profile output
 #------------------------------------------------------------------------------
 
-CLEAN = *.o *.obj *.ln *.bb *.bbg *.da *.tcov *.gcov gmon.out *.bak *.d
+CLEAN = *.o *.obj *.ln *.bb *.bbg *.da *.tcov *.gcov gmon.out *.bak *.d *.a *.def
diff -urN -x *.o -x *.dll -x *.a -x *.exe -x *.def -x *.out SuiteSparse-orig/UMFPACK/Source/GNUmakefile SuiteSparse/UMFPACK/Source/GNUmakefile
--- SuiteSparse-orig/UMFPACK/Source/GNUmakefile	Tue May  2 15:36:45 2006
+++ SuiteSparse/UMFPACK/Source/GNUmakefile	Fri Jun 15 21:53:59 2007
@@ -2,7 +2,7 @@
 # UMFPACK Makefile for compiling on Unix systems (for GNU Make)
 #-------------------------------------------------------------------------------
 
-default: ../Lib/libumfpack.a
+default: ../Lib/libumfpack.a ../Lib/umfpack.dll
 
 include ../../UFconfig/UFconfig.mk
 
@@ -246,12 +246,20 @@
 	$(AR) ../Lib/libumfpack.a $^
 	- $(RANLIB) ../Lib/libumfpack.a
 
+../Lib/umfpack.dll : ../Lib/libumfpack.a
+	$(CC) -shared $(SHARED_FLAGS) -o $@ -L../../AMD/Lib \
+		-Wl,--whole-archive $< \
+		-Wl,--out-implib,$(basename $<).dll.a \
+		-Wl,--output-def,$(basename $<).def \
+		-Wl,--no-whole-archive -lamd $(BLAS)
+	$(STRIP) $(STRIP_FLAGS) $@
+
 #-------------------------------------------------------------------------------
 # Remove all but the files in the original distribution
 #-------------------------------------------------------------------------------
 
 purge: clean
-	- $(RM) ../Lib/libumfpack.a
+	- $(RM) ../Lib/libumfpack.a ../Lib/umfpack.dll ../Lib/libumfpack.dll.a
 
 clean:
 	- $(RM) $(CLEAN)
