ensure quoted pager binary

diff -r ee6e50ab358b src/ChangeLog
--- a/src/ChangeLog	Thu Aug 20 20:27:21 2009 +0200
+++ b/src/ChangeLog	Thu Aug 20 20:28:01 2009 +0200
@@ -317,6 +317,11 @@
 
 	* variables.cc (do_who): Only output symbols with a defined value.
 
+2009-05-24  Benjamin Lindner <lindnerb@users.sourceforge.net>
+
+	* pager.cc: enclose VPAGER in double quotes and enclose
+	total pager command string in double quotes on mingw platform
+
 2009-05-22  Michael Goffioul  <michael.goffioul@gmail.com>
 
 	* toplev.h (main_loop): Tag with OCTINTERP_API.
diff -r ee6e50ab358b src/pager.cc
--- a/src/pager.cc	Thu Aug 20 20:27:21 2009 +0200
+++ b/src/pager.cc	Thu Aug 20 20:28:01 2009 +0200
@@ -132,11 +132,40 @@
 static std::string
 pager_command (void)
 {
-  std::string cmd = VPAGER;
+  std::string cmd = "\"" + VPAGER + "\"";
 
   if (! (cmd.empty () || VPAGER_FLAGS.empty ()))
+  {
     cmd += " " + VPAGER_FLAGS;
 
+#ifdef __MINGW32__
+  // the documentation of CMD.EXE tells:
+  //
+  //  If /C or /K is specified, then the remainder of the command line after
+  //  the switch is processed as a command line, where the following logic is
+  //  used to process quote (") characters:
+  //  
+  //      1.  If all of the following conditions are met, then quote characters
+  //          on the command line are preserved:
+  //  
+  //          - no /S switch
+  //          - exactly two quote characters
+  //          - no special characters between the two quote characters,
+  //            where special is one of: &<>()@^|
+  //          - there are one or more whitespace characters between the
+  //            the two quote characters
+  //          - the string between the two quote characters is the name
+  //            of an executable file.
+  //  
+  //      2.  Otherwise, old behavior is to see if the first character is
+  //          a quote character and if so, strip the leading character and
+  //          remove the last quote character on the command line, preserving
+  //          any text after the last quote character.
+    
+    cmd = "\"" + cmd + "\"";
+#endif
+  }
+  
   return cmd;
 }
 
