disable SIGINT handler for console build

diff -r 329257da0ade src/plot.c
--- a/src/plot.c	Mon Aug 31 09:13:24 2009 +0200
+++ b/src/plot.c	Mon Aug 31 09:13:38 2009 +0200
@@ -684,7 +684,11 @@
     setmatherr(purec_matherr);
 #endif
 
+#if defined(WGP_CONSOLE)
+    (void) signal(SIGINT, SIG_IGN);
+#else
     (void) signal(SIGINT, (sigfunc) inter);
+#endif
 
 #ifdef SIGPIPE
     /* ignore pipe errors, this might happen with set output "|head" */
