enable support for CAIRO terminals without wxt term

diff -r 6418d4f24300 config/makefile.mgw
--- a/config/makefile.mgw	Mon Sep 21 10:58:58 2009 +0200
+++ b/config/makefile.mgw	Mon Sep 21 10:59:33 2009 +0200
@@ -83,8 +83,8 @@
 # Cairo/Pango terminals
 # Uncomment the following lines to select the pngcairo and pdfcairo
 # terminals
-#CAIROTERMS=1
-#CAIROLIBS=1
+CAIROTERMS=1
+CAIROLIBS=1
 #
 
 # WXWIDGETS interactive terminal
