add override directive to CFLAGS CXXFLAGS LDFLAGS and RCFLAGS

diff -r 73ca544272a4 -r 5ba130e706ff config/makefile.mgw
--- a/config/makefile.mgw	Tue Jan 12 21:45:04 2010 +0100
+++ b/config/makefile.mgw	Tue Jan 12 21:46:47 2010 +0100
@@ -180,11 +180,11 @@
 CP = cp -p
 
 ifdef DEBUG
-  CFLAGS += -g -DDEBUG
-  LDFLAGS += -g
+  override CFLAGS += -g -DDEBUG
+  override LDFLAGS += -g
 else
-  CFLAGS += -O2
-  LDFLAGS += -s
+  override CFLAGS += -O2
+  override LDFLAGS += -s
 endif
 
 LDFLAGS2 = -mwindows
@@ -202,11 +202,11 @@
 TERMFLAGS = -DGNUPLOT_PS_DIR=\"$(GNUPLOT_PS_DIR)\" -DGNUPLOT_JS_DIR=\"$(GNUPLOT_JS_DIR)\" -I$(TOP)/term
 TERMLIBS = 
 
-CFLAGS += -I$(TOP) -I$(TOP)/src -I. -D_Windows -DHAVE_CONFIG_H\
+override CFLAGS += -I$(TOP) -I$(TOP)/src -I. -D_Windows -DHAVE_CONFIG_H\
 	$(OPTS)
 
 ifdef MOUSE
-  CFLAGS += -DUSE_MOUSE=1 -DWIN_IPC
+  override CFLAGS += -DUSE_MOUSE=1 -DWIN_IPC
 endif
 
 ifdef NEWGD
@@ -215,27 +215,27 @@
 endif
 
 ifdef GD
-  CFLAGS += -DHAVE_LIBGD
+  override CFLAGS += -DHAVE_LIBGD
   TERMLIBS += -lgd
 endif
 
 ifdef PNG
-  CFLAGS += -DHAVE_LIBPNG
+  override CFLAGS += -DHAVE_LIBPNG
   ifndef GDAUTOCONFIGLIBS
     TERMLIBS += -lpng -lz
   endif
 endif
 
 ifdef NEWGD
-  CFLAGS += -DHAVE_GD_GIF -DGIF_ANIMATION -DHAVE_GD_PNG
+  override CFLAGS += -DHAVE_GD_GIF -DGIF_ANIMATION -DHAVE_GD_PNG
 ifdef JPEG
-  CFLAGS += -DHAVE_GD_JPEG
+  override CFLAGS += -DHAVE_GD_JPEG
   ifndef GDAUTOCONFIGLIBS
     TERMLIBS += -ljpeg
   endif
 endif
 ifdef FREETYPE
-  CFLAGS += -DHAVE_GD_TTF
+  override CFLAGS += -DHAVE_GD_TTF
   ifndef GDAUTOCONFIGLIBS
     TERMLIBS += -lfreetype
   endif
@@ -246,7 +246,7 @@
 endif
 
 ifdef PDF
-  CFLAGS += -DHAVE_LIBPDF
+  override CFLAGS += -DHAVE_LIBPDF
 ifdef PDF_NOTIFF
   TERMLIBS += -lpdf_notiff
 else
@@ -261,7 +261,7 @@
 endif
 endif
 ifdef NEED_PNG_W
-  CFLAGS += -DHAVE_LIBPNG
+  override CFLAGS += -DHAVE_LIBPNG
   TERMLIBS += -lpng-w -lz-w
 endif
 endif
@@ -276,12 +276,12 @@
 endif
 
 ifdef CAIROTERMS
-	CFLAGS += -DHAVE_CAIROPDF
+	override CFLAGS += -DHAVE_CAIROPDF
 endif
 
 ifdef WXT
-	CFLAGS += -DWXWIDGETS
-	CXXFLAGS += $(shell wx-config --cxxflags)
+	override CFLAGS += -DWXWIDGETS
+	override CXXFLAGS += $(shell wx-config --cxxflags)
 	WX_LIBS = $(shell wx-config --libs | sed -e "s+-Wl,--subsystem,windows++g" -e "s+-mwindows++g")
 	WX_OBJS = wxt_gui.o
 	WX_LOCATION = $(shell wx-config --cppflags)
@@ -290,12 +290,12 @@
 ifdef GNU_RC
   # RC = d:/cygnus/cygwin-b20/H-i586-cygwin32/bin/windres
   RC  = $(GCCPATH)windres
-  RCFLAGS = --include-dir /mingw32/include \
+  override RCFLAGS = --include-dir /mingw32/include \
 	--include-dir=$(TOP)/src/win \
 	--define __WIN32__ --define __WIN95__ --define MSRC \
 	--define __GNUWIN32__ --use-temp-file
   ifdef WXT
-	RCFLAGS += --define WXWIDGETS $(WX_LOCATION)
+	override RCFLAGS += --define WXWIDGETS $(WX_LOCATION)
   endif
   RCOUT = wgplt_res.o
   RES2COFF = echo wgplt_res.o
