local changes to makefile to support wxwidgets terminal

diff -r e97097cda53f config/makefile.mgw
--- a/config/makefile.mgw	Tue Dec 01 09:44:26 2009 +0100
+++ b/config/makefile.mgw	Tue Dec 01 09:45:57 2009 +0100
@@ -284,6 +284,7 @@
 	CXXFLAGS += $(shell wx-config --cxxflags)
 	WX_LIBS = $(shell wx-config --libs | sed -e "s+-Wl,--subsystem,windows++g" -e "s+-mwindows++g")
 	WX_OBJS = wxt_gui.o
+	WX_LOCATION = $(shell wx-config --cppflags)
 endif
 
 ifdef GNU_RC
@@ -294,7 +295,7 @@
 	--define __WIN32__ --define __WIN95__ --define MSRC \
 	--define __GNUWIN32__ --use-temp-file
   ifdef WXT
-	RCFLAGS += --define WXWIDGETS --include-dir=$(WX_LOCATION)
+	RCFLAGS += --define WXWIDGETS $(WX_LOCATION)
   endif
   RCOUT = wgplt_res.o
   RES2COFF = echo wgplt_res.o
@@ -302,7 +303,7 @@
   RC = $(MSSDK)/bin/rc
   RCFLAGS = -v -i$(MSSDK)/include -iwin -dWIN32 -dMSRC
   ifdef WXT
-	RCFLAGS += -dWXWIDGETS -i$(WX_LOCATION)
+	RCFLAGS += -dWXWIDGETS $(WX_LOCATION)
   endif
   RCOUT = -fowgnuplot.res
   RES2COFF = res2coff -i wgnuplot.res -o wgplt_res.o
@@ -349,6 +350,7 @@
 vpath	%.def	$(TOP)/src
 vpath	%.rc	$(TOP)/src
 vpath	%.trm	$(TOP)/term
+vpath	%.cpp	$(TOP)/src
 
 #
 # Create config.h
@@ -405,7 +407,7 @@
 wgnuplot.mnu: $(TOP)/src/win/wgnuplot.mnu
 	$(CP) $^ $@
 
-wxt_gui.$(O): wxterminal/wxt_gui.cpp wxterminal/wxt_gui.h $(WX_LOCATION)
+wxt_gui.$(O): wxterminal/wxt_gui.cpp wxterminal/wxt_gui.h
 	$(CXX) -c $(CFLAGS) $(CXXFLAGS) $(CAIRO_CFLAGS) $<
 
 gp_cairo.$(O): wxterminal/gp_cairo.c wxterminal/gp_cairo.h
