add override directive in makefile.mgw

diff -r 9f4276db4d7c config/makefile.mgw
--- a/config/makefile.mgw	Thu Oct 01 20:37:15 2009 +0200
+++ b/config/makefile.mgw	Thu Oct 01 20:38:32 2009 +0200
@@ -107,7 +107,7 @@
 # Do you want some special optimization?
 # -mpentium means optimise for Pentium processor
 # -mpentiumpro means optimize for Pentium II and Pro procesors
-CFLAGS += $(GCC_ARCH_FLAGS) -Wall
+override CFLAGS += $(GCC_ARCH_FLAGS) -Wall
 
 
 # To compile the .hlp file you need hcw either out of Microsoft SDK or MS Help
@@ -171,11 +171,11 @@
 CP = cp -p
 
 ifdef DEBUG
-  CFLAGS += -g
-  LDFLAGS += -g
+  override CFLAGS += -g
+  override LDFLAGS += -g
 else
-  CFLAGS += -O2
-  LDFLAGS += -s
+  override CFLAGS += -O2
+  override LDFLAGS += -s
 endif
 
 LDFLAGS2 = -mwindows
@@ -193,11 +193,11 @@
 TERMFLAGS = -DGNUPLOT_PS_DIR=\"$(GNUPLOT_PS_DIR)\" -I$(TOP)/term
 TERMLIBS = 
 
-CFLAGS += -I$(TOP) -I$(TOP)/src -I. -D_Windows -DHAVE_CONFIG_H\
+override CFLAGS += -I$(TOP) -I$(TOP)/src -I. -D_Windows -DHAVE_CONFIG_H\
 	$(OPTS)
 
 ifdef MOUSE
-  CFLAGS += -DUSE_MOUSE=1 -DWIN_IPC
+  override CFLAGS += -DUSE_MOUSE=1 -DWIN_IPC
 endif
 
 ifdef NEWGD
@@ -206,27 +206,27 @@
 endif
 
 ifdef GD
-  CFLAGS += -DHAVE_LIBGD
+  override CFLAGS += -DHAVE_LIBGD
   TERMLIBS += -lgd
 endif
 
 ifdef PNG
-  CFLAGS += -DHAVE_LIBPNG
+  override CFLAGS += -DHAVE_LIBPNG
   ifndef GDAUTOCONFIGLIBS
     TERMLIBS += -lpng -lz
   endif
 endif
 
 ifdef NEWGD
-  CFLAGS += -DHAVE_GD_GIF -DGIF_ANIMATION -DHAVE_GD_PNG
+  override CFLAGS += -DHAVE_GD_GIF -DGIF_ANIMATION -DHAVE_GD_PNG
 ifdef JPEG
-  CFLAGS += -DHAVE_GD_JPEG
+  override CFLAGS += -DHAVE_GD_JPEG
   ifndef GDAUTOCONFIGLIBS
     TERMLIBS += -ljpeg
   endif
 endif
 ifdef FREETYPE
-  CFLAGS += -DHAVE_GD_TTF
+  override CFLAGS += -DHAVE_GD_TTF
   ifndef GDAUTOCONFIGLIBS
     TERMLIBS += -lfreetype
   endif
@@ -237,7 +237,7 @@
 endif
 
 ifdef PDF
-  CFLAGS += -DHAVE_LIBPDF
+  override CFLAGS += -DHAVE_LIBPDF
 ifdef PDF_NOTIFF
   TERMLIBS += -lpdf_notiff
 else
@@ -252,7 +252,7 @@
 endif
 endif
 ifdef NEED_PNG_W
-  CFLAGS += -DHAVE_LIBPNG
+  override CFLAGS += -DHAVE_LIBPNG
   TERMLIBS += -lpng-w -lz-w
 endif
 endif
@@ -268,8 +268,8 @@
 endif
 
 ifdef WXT
-	CFLAGS += -DWXWIDGETS -DHAVE_CAIROPDF
-	CXXFLAGS += $(shell wx-config --cxxflags)
+	override CFLAGS += -DWXWIDGETS -DHAVE_CAIROPDF
+	override CXXFLAGS += $(shell wx-config --cxxflags)
 	CAIRO_CFLAGS = $(shell pkg-config --cflags cairo)
 	CAIRO_LIBS = $(shell pkg-config --libs cairo)
 	PANGO_CFLAGS = $(shell pkg-config --cflags pangocairo)
@@ -282,20 +282,20 @@
 ifdef GNU_RC
   # RC = d:/cygnus/cygwin-b20/H-i586-cygwin32/bin/windres
   RC  = $(GCCPATH)windres
-  RCFLAGS = --include-dir /mingw32/include \
+  override RCFLAGS = --include-dir /mingw32/include \
 	--include-dir=$(TOP)/src/win \
 	--define __WIN32__ --define __WIN95__ --define MSRC \
 	--define __GNUWIN32__ --use-temp-file --preprocessor=$(CPP)
   ifdef WXT
-	RCFLAGS += --define WXWIDGETS $(WX_LOCATION)
+	override RCFLAGS += --define WXWIDGETS $(WX_LOCATION)
   endif
   RCOUT = wgplt_res.o
   RES2COFF = echo wgplt_res.o
 else
   RC = $(MSSDK)/bin/rc
-  RCFLAGS = -v -i$(MSSDK)/include -iwin -dWIN32 -dMSRC
+  override RCFLAGS = -v -i$(MSSDK)/include -iwin -dWIN32 -dMSRC
   ifdef WXT
-	RCFLAGS += -dWXWIDGETS $(WX_LOCATION)
+	override RCFLAGS += -dWXWIDGETS $(WX_LOCATION)
   endif
   RCOUT = -fowgnuplot.res
   RES2COFF = res2coff -i wgnuplot.res -o wgplt_res.o
