enable support for CAIRO terminals without wxt term

diff -r 337757fb9f97 config/makefile.mgw
--- a/config/makefile.mgw	Wed Aug 26 16:09:07 2009 +0200
+++ b/config/makefile.mgw	Wed Aug 26 16:09:40 2009 +0200
@@ -57,6 +57,7 @@
 NEWGD=1
 JPEG=1
 FREETYPE=1
+CAIRO=1
 #GDAUTOCONFIGLIBS=1
 
 # PDF device driver
@@ -255,6 +256,16 @@
 endif
 endif
 
+ifdef CAIRO
+	CFLAGS += -DHAVE_CAIROPDF
+	CAIRO_CFLAGS = $(shell pkg-config --cflags cairo)
+	CAIRO_LIBS = $(shell pkg-config --libs cairo)
+	PANGO_CFLAGS = $(shell pkg-config --cflags pangocairo)
+	PANGO_LIBS = $(shell pkg-config --libs pangocairo)
+	WX_OBJS =  gp_cairo.o gp_cairo_helpers.o
+	TERMFLAGS += $(CAIRO_CFLAGS) $(PANGO_CFLAGS)
+endif
+
 ifdef WXT
 	CFLAGS += -DWXWIDGETS
 	CXXFLAGS += $(shell wx-config --cxxflags)
@@ -392,7 +403,7 @@
 	$(CC) -c $(CFLAGS) $(CXXFLAGS) $(CAIRO_CFLAGS) $(PANGO_CFLAGS) $<
 
 gp_cairo_helpers.$(O): wxterminal/gp_cairo_helpers.c wxterminal/gp_cairo_helpers.h
-	$(CC) -c $(CFLAGS) wxterminal/gp_cairo_helpers.c
+	$(CC) -c $(CFLAGS) $<
 
 # extract icons from wgnuplot.rc
 texticon.ico: grpicon.ico
