in makefile: local build configuration

diff --git a/config/makefile.mgw b/config/makefile.mgw
--- a/config/makefile.mgw
+++ b/config/makefile.mgw
@@ -20,7 +20,7 @@
 #
 
 # Specify the path to the sources (top directory)
-SRCDIR=..
+SRCDIR=@SRCDIR@
 
 # Choose target to compile:
 # 1. wgnuplot.exe: The usual Windows GUI for input commands.
@@ -35,7 +35,7 @@
 # used when gnuplot.exe was not available on Windows. Nowadays, pgnuplot.exe
 # is deprecated -- plese use gnuplot.exe instead.
 
-TARGET=wgnuplot.exe
+TARGET=@TARGET@
 #TARGET=wgnuplot_pipes.exe
 #TARGET=gnuplot.exe
 
@@ -106,7 +106,7 @@
 
 # Where to install the PostScript prologue files, relatively to the executable
 # wgnuplot.exe
-GNUPLOT_PS_DIR = share/PostScript
+GNUPLOT_PS_DIR = share\\\\gnuplot\\\\PostScript
 
 # Similarly for javascript files for the canvas terminal:
 GNUPLOT_JS_DIR = share/js
@@ -124,7 +124,7 @@
 # Do you want some special optimization?
 # -mpentium means optimise for Pentium processor
 # -mpentiumpro means optimize for Pentium II and Pro procesors
-CFLAGS =
+CFLAGS += $(GCC_ARCH_FLAGS) -Wall
 
 
 # To compile the .hlp file you need hcw either out of Microsoft SDK or MS Help
@@ -147,6 +147,9 @@
 # GNU sort
 GNUSORT = /bin/sort
 
+# GNU sed
+SED=/usr/bin/sed
+
 #
 # *************** End of Configuration section *************************
 #
@@ -178,10 +181,8 @@
 endif
 
 
-CC = $(GCCPATH)gcc
-LD = $(GCCPATH)gcc
-CXX = $(GCCPATH)g++
-LDX = $(GCCPATH)g++
+LD = $(CC)
+LDX = $(CXX)
 
 RM = rm -f
 CP = cp -p
@@ -201,7 +202,7 @@
 endif
 
 ifdef CONSOLE
-  OPTS += -DWGP_CONSOLE
+  OPTS += -DWGP_CONSOLE -DVOLATILE_REFRESH
   LDFLAGS2 = -mconsole
 endif
 
@@ -307,7 +308,7 @@
   override RCFLAGS = --include-dir /mingw32/include \
 	--include-dir=$(TOP)/src/win \
 	--define __WIN32__ --define __WIN95__ --define MSRC \
-	--define __GNUWIN32__ --use-temp-file
+	--define __GNUWIN32__ --use-temp-file --preprocessor=$(CPP)
   ifdef WXT
 	override RCFLAGS += --define WXWIDGETS $(WX_LOCATION)
   endif
@@ -473,7 +474,7 @@
 	@for e in `egrep "^[ 	]*START_HELP" $^ |\
 	     $(GNUSORT) -f -t':' -k2` ; do \
 	  f=`echo $$e |cut -d\: -f1` ; s=`echo $$e | cut -d\: -f2` ;\
-	  sed -n "/^[ 	]*$$s/,/^[ 	]*END_HELP/p" $$f ; \
+	  $(SED) -n "/^[ 	]*$$s/,/^[ 	]*END_HELP/p" $$f ; \
 	done >$@
 else
 # sequence of terminals according to "ls term/*.trm":
@@ -481,7 +482,7 @@
 	@echo "Building allterm.h"
 	@cat $(T)*.trm > allterm.c
 	$(CPP) $(CFLAGS) -I$(T:/=) -DTERM_DRIVER_H -DTERM_HELP allterm.c | \
-	  sed '/^ *$$/d;/^#/d' > allterm.h
+	  $(SED) '/^ *$$/d;/^#/d' > allterm.h
 	@rm -f allterm.c
 endif
 doc2tex.exe: $(D)doc2tex.c $(D)termdoc.c allterm.h
