gnuplot/octave version

diff -r 01feefdeb53e src/version.c
--- a/src/version.c	Tue Jan 12 21:47:13 2010 +0100
+++ b/src/version.c	Mon Jan 18 21:44:27 2010 +0100
@@ -40,7 +40,7 @@
 
 
 const char gnuplot_version[] = "4.4";
-const char gnuplot_patchlevel[] = "0-rc1";
+const char gnuplot_patchlevel[] = "0-rc1 octave-mingw32";
 const char gnuplot_date[] = "Sat Nov 28 14:34:01 PST 2009";
 const char gnuplot_copyright[] = "Copyright (C) 1986-1993, 1998, 2004, 2007-2009";
 
