set origin of postscript output to 0,0 (no offset)

diff -r a26efd09d6e7 term/post.trm
--- a/term/post.trm	Wed Aug 26 15:45:12 2009 +0200
+++ b/term/post.trm	Wed Aug 26 15:46:15 2009 +0200
@@ -192,7 +192,7 @@
 } ps_params_t;
 
 #define POST_PARAMS_DEFAULT { \
-    PSTERM_POSTSCRIPT, 50, 50, \
+    PSTERM_POSTSCRIPT, 0, 0, \
     PSTERM_LANDSCAPE, FALSE, FALSE, FALSE, FALSE, 1.0, 1.0, FALSE, \
     FALSE, FALSE, FALSE, NULL, "Helvetica", 14, FALSE, FALSE, 2000, 0.003, \
     FALSE, TRUE, FALSE \
@@ -202,7 +202,7 @@
 static const ps_params_t post_params_default = POST_PARAMS_DEFAULT;
 
 #define EPSLATEX_PARAMS_DEFAULT { \
-    PSTERM_EPSLATEX, 50, 50, \
+    PSTERM_EPSLATEX, 0, 0, \
     PSTERM_EPS, FALSE, FALSE, TRUE, FALSE, 1.0, 1.0, FALSE, \
     FALSE, FALSE, FALSE, NULL, "", 11, TRUE, FALSE, 2000, 0.003, \
     FALSE, FALSE, FALSE \
