set origin of postscript output to 0,0 (no offset)

diff -r 721248ae0dda term/post.trm
--- a/term/post.trm	Tue Jan 27 10:10:54 2009 +0100
+++ b/term/post.trm	Tue Jan 27 10:11:41 2009 +0100
@@ -193,7 +193,7 @@
 } ps_params_t;
 
 #define POST_PARAMS_DEFAULT { \
-    PSTERM_POSTSCRIPT, 50, 50, \
+    PSTERM_POSTSCRIPT, 0, 0, \
     PSTERM_LANDSCAPE, FALSE, FALSE, FALSE, FALSE, 1.0, 1.0, FALSE, \
     FALSE, FALSE, NULL, "Helvetica", 14, FALSE, FALSE, 2000, 0.003, \
     FALSE, TRUE \
@@ -203,7 +203,7 @@
 static const ps_params_t post_params_default = POST_PARAMS_DEFAULT;
 
 #define EPSLATEX_PARAMS_DEFAULT { \
-    PSTERM_EPSLATEX, 50, 50, \
+    PSTERM_EPSLATEX, 0, 0, \
     PSTERM_EPS, FALSE, FALSE, TRUE, FALSE, 1.0, 1.0, FALSE, \
     FALSE, FALSE, NULL, "", 11, TRUE, FALSE, 2000, 0.003, \
     FALSE, FALSE \
