diff -urN -x '*.exe' -x '*.dll' -x '*.o' -x '*.a' -x '*.bak' -x '~' -x .hg -x '*.orig' cln-1.3.0-orig/src/Makefile.in cln-1.3.0/src/Makefile.in
--- cln-1.3.0-orig/src/Makefile.in	2009-10-26 18:04:43 +0100
+++ cln-1.3.0/src/Makefile.in	2009-10-26 18:03:01 +0100
@@ -480,7 +480,7 @@
 top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-lib_LTLIBRARIES = libcln.la
+lib_LTLIBRARIES = libcln.la cln.dll
 AM_CPPFLAGS = -I$(top_srcdir)/include -I$(top_srcdir)/src -I$(top_builddir)/include -I$(top_builddir)/src
 DEFAULT_INCLUDES = -I.@am__isrc@ -I$(top_builddir)/autoconf
 libcln_la_LDFLAGS = -version-info $(LT_VERSION_INFO)
@@ -1521,6 +1521,14 @@
 libcln.la: $(libcln_la_OBJECTS) $(libcln_la_DEPENDENCIES) 
 	$(libcln_la_LINK) -rpath $(libdir) $(libcln_la_OBJECTS) $(libcln_la_LIBADD) $(LIBS)
 
+cln.dll : libcln.la
+	$(CXX) -shared -o $@ \
+	$(LDFLAGS) \
+	-Wl,--whole-archive .libs/libcln.a \
+	-Wl,--no-whole-archive \
+	-Wl,-out-implib=libcln.dll.a \
+	$(LIBS) 
+
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
 
