diff -urN -x *.exe -x *.dll -x *.o -x *.a -x *.bak gsl-1.10-orig/configure gsl-1.10/configure
--- gsl-1.10-orig/configure	Thu Sep 13 17:41:52 2007
+++ gsl-1.10/configure	Mon Jan 21 22:30:32 2008
@@ -7632,7 +7632,7 @@
       ;;
     mingw*)
       # MinGW DLLs use traditional 'lib' prefix
-      soname_spec='${libname}`echo ${release} | $SED -e 's/[.]/-/g'`${versuffix}${shared_ext}'
+      soname_spec='`echo ${libname} | sed -e 's/^lib//'`${shared_ext}'
       sys_lib_search_path_spec=`$CC -print-search-dirs | grep "^libraries:" | $SED -e "s/^libraries://" -e "s,=/,/,g"`
       if echo "$sys_lib_search_path_spec" | grep ';[c-zC-Z]:/' >/dev/null; then
         # It is most probably a Windows format PATH printed by
