diff -urN -x '*.exe' -x '*.dll' -x '*.o' -x '*.a' -x '*.bak' gsl-1.11-orig/configure gsl-1.11/configure
--- gsl-1.11-orig/configure	2008-03-29 17:23:19 +0100
+++ gsl-1.11/configure	2008-07-26 11:08:26 +0200
@@ -8929,7 +8929,7 @@
       ;;
     mingw*)
       # MinGW DLLs use traditional 'lib' prefix
-      soname_spec='${libname}`echo ${release} | $SED -e 's/[.]/-/g'`${versuffix}${shared_ext}'
+      soname_spec='`echo ${libname} | sed -e 's/^lib//'`${shared_ext}'
       sys_lib_search_path_spec=`$CC -print-search-dirs | $GREP "^libraries:" | $SED -e "s/^libraries://" -e "s,=/,/,g"`
       if $ECHO "$sys_lib_search_path_spec" | $GREP ';[c-zC-Z]:/' >/dev/null; then
         # It is most probably a Windows format PATH printed by
