diff -urN -x *.exe -x *.dll -x *.o -x *.a -x *.bak -x *.def lapack-lite-3.1.1-orig/INSTALL/Makefile lapack-lite-3.1.1/INSTALL/Makefile
--- lapack-lite-3.1.1-orig/INSTALL/Makefile	Mon Feb 26 19:19:47 2007
+++ lapack-lite-3.1.1/INSTALL/Makefile	Sat Dec  8 22:02:24 2007
@@ -1,5 +1,9 @@
 include ../make.inc
 
+SRCDIR = $(SRCTOP)/INSTALL
+
+vpath %.f $(SRCDIR)
+
 .SUFFIXES : .o .f 
 all:  testlsame testslamch testdlamch testsecond testdsecnd testieee testversion 
 
diff -urN -x *.exe -x *.dll -x *.o -x *.a -x *.bak -x *.def lapack-lite-3.1.1-orig/Makefile lapack-lite-3.1.1/Makefile
--- lapack-lite-3.1.1-orig/Makefile	Mon Feb 26 19:20:49 2007
+++ lapack-lite-3.1.1/Makefile	Sat Dec  8 21:57:19 2007
@@ -22,6 +22,12 @@
 
 lapacklib:	lapack_install
 	( cd SRC; $(MAKE) )
+	$(F77) -shared $(FLDFLAGS) -o lapack.dll \
+	  -Wl,--export-all-symbols \
+	  -Wl,--out-implib=lapack.dll.a \
+	  -Wl,--whole-archive lapack.a \
+	  -Wl,--no-whole-archive -lblas
+	$(STRIP) $(STRIP_FLAGS) lapack.dll
 
 tmglib:
 	( cd TESTING/MATGEN; $(MAKE) )
diff -urN -x *.exe -x *.dll -x *.o -x *.a -x *.bak -x *.def lapack-lite-3.1.1-orig/SRC/Makefile lapack-lite-3.1.1/SRC/Makefile
--- lapack-lite-3.1.1-orig/SRC/Makefile	Mon Feb 26 19:20:07 2007
+++ lapack-lite-3.1.1/SRC/Makefile	Sat Dec  8 22:02:55 2007
@@ -1,5 +1,9 @@
 include ../make.inc
 
+SRCDIR = $(SRCTOP)/SRC
+
+vpath %.f $(SRCDIR)
+
 #######################################################################
 #  This is the makefile to create a library for LAPACK.
 #  The files are organized as follows:
diff -urN -x *.exe -x *.dll -x *.o -x *.a -x *.bak -x *.def lapack-lite-3.1.1-orig/make.inc lapack-lite-3.1.1/make.inc
--- lapack-lite-3.1.1-orig/make.inc	Thu Jan  1 01:00:00 1970
+++ lapack-lite-3.1.1/make.inc	Sat Dec  8 22:22:44 2007
@@ -0,0 +1,59 @@
+####################################################################
+#  LAPACK make include file.                                       #
+#  LAPACK, Version 3.1.1                                           #
+#  February 2007                                                   #
+####################################################################
+#
+# See the INSTALL/ directory for more examples.
+#
+SHELL = /bin/sh
+#
+#  The machine (platform) identifier to append to the library names
+#
+PLAT = 
+#  
+#  Modify the FORTRAN and OPTS definitions to refer to the
+#  compiler and desired compiler options for your machine.  NOOPT
+#  refers to the compiler options desired when NO OPTIMIZATION is
+#  selected.  Define LOADER and LOADOPTS to refer to the loader and 
+#  desired load options for your machine.
+#
+FORTRAN  = $(F77) 
+OPTS     = -funroll-all-loops -O3
+DRVOPTS  = $(OPTS)
+NOOPT    =
+LOADER   = $(F77)
+LOADOPTS = 
+#
+# Timer for the SECOND and DSECND routines
+#
+# Default : SECOND and DSECND will use a call to the EXTERNAL FUNCTION ETIME
+# TIMER    = EXT_ETIME
+# For RS6K : SECOND and DSECND will use a call to the EXTERNAL FUNCTION ETIME_
+# TIMER    = EXT_ETIME_
+# For gfortran compiler: SECOND and DSECND will use a call to the INTERNAL FUNCTION ETIME
+# TIMER    = INT_ETIME 
+# If your Fortran compiler does not provide etime (like Nag Fortran Compiler, etc...)
+# SECOND and DSECND will use a call to the INTERNAL FUNCTION CPU_TIME 
+TIMER    = INT_CPU_TIME
+# If neither of this works...you can use the NONE value... In that case, SECOND and DSECND will always return 0
+# TIMER     = NONE
+#
+#  The archiver and the flag(s) to use when building archive (library)
+#  If you system has no ranlib, set RANLIB = echo.
+#
+ARCH     = ar
+ARCHFLAGS= cr
+RANLIB   = ranlib
+#
+#  The location of the libraries to which you will link.  (The 
+#  machine-specific, optimized BLAS library should be used whenever
+#  possible.)
+#
+BLASLIB      = -lblas
+LAPACKLIB    = lapack$(PLAT).a
+TMGLIB       = tmglib$(PLAT).a
+EIGSRCLIB    = eigsrc$(PLAT).a
+LINSRCLIB    = linsrc$(PLAT).a
+
+SRCTOP = @SRCDIR@
