diff -urN -x '*.exe' -x '*.dll' -x '*.o' -x '*.a' -x '*.bak' freetype-2.3.7-orig/builds/unix/freetype-config.in freetype-2.3.7/builds/unix/freetype-config.in
--- freetype-2.3.7-orig/builds/unix/freetype-config.in	2008-06-09 21:25:37 +0000
+++ freetype-2.3.7/builds/unix/freetype-config.in	2008-11-21 08:52:43 +0000
@@ -128,7 +128,7 @@
 fi
 
 if test "$echo_cflags" = "yes" ; then
-  cflags="-I$includedir/freetype2"
+  cflags="-I$includedir"
   if test "$includedir" != "/usr/include" ; then
     echo $cflags -I$includedir
   else
diff -urN -x '*.exe' -x '*.dll' -x '*.o' -x '*.a' -x '*.bak' freetype-2.3.7-orig/builds/unix/freetype2.in freetype-2.3.7/builds/unix/freetype2.in
--- freetype-2.3.7-orig/builds/unix/freetype2.in	2006-10-12 04:51:08 +0000
+++ freetype-2.3.7/builds/unix/freetype2.in	2008-11-21 08:51:15 +0000
@@ -8,4 +8,4 @@
 Version: @ft_version@
 Requires:
 Libs: -L${libdir} -lfreetype @LIBZ@ @FT2_EXTRA_LIBS@
-Cflags: -I${includedir}/freetype2 -I${includedir}
+Cflags: -I${includedir}
