diff -urN -x '*.exe' -x '*.dll' -x '*.o' -x '*.a' -x '*.bak' freetype-2.3.7-orig/builds/unix/freetype-config.in freetype-2.3.7/builds/unix/freetype-config.in
--- freetype-2.3.7-orig/builds/unix/freetype-config.in	2008-06-09 23:25:37 +0200
+++ freetype-2.3.7/builds/unix/freetype-config.in	2009-01-11 21:56:36 +0100
@@ -128,12 +128,8 @@
 fi
 
 if test "$echo_cflags" = "yes" ; then
-  cflags="-I$includedir/freetype2"
-  if test "$includedir" != "/usr/include" ; then
-    echo $cflags -I$includedir
-  else
+  cflags=""
     echo $cflags
-  fi
 fi
 
 if test "$echo_libs" = "yes" ; then
@@ -142,11 +138,7 @@
     eval "rpath=\"$hardcode_libdir_flag_spec\""
   fi
   libs="-lfreetype @LIBZ@ @FT2_EXTRA_LIBS@"
-  if test "$libdir" != "/usr/lib" && test "$libdir" != "/usr/lib64"; then
-    echo -L$libdir $rpath $libs
-  else
     echo $libs
-  fi
 fi
 
 if test "$echo_libtool" = "yes" ; then
diff -urN -x '*.exe' -x '*.dll' -x '*.o' -x '*.a' -x '*.bak' freetype-2.3.7-orig/builds/unix/freetype2.in freetype-2.3.7/builds/unix/freetype2.in
--- freetype-2.3.7-orig/builds/unix/freetype2.in	2006-10-12 06:51:08 +0200
+++ freetype-2.3.7/builds/unix/freetype2.in	2008-11-30 13:42:32 +0100
@@ -8,4 +8,4 @@
 Version: @ft_version@
 Requires:
 Libs: -L${libdir} -lfreetype @LIBZ@ @FT2_EXTRA_LIBS@
-Cflags: -I${includedir}/freetype2 -I${includedir}
+Cflags: -I${includedir}
