diff -urN -x *.exe -x *.dll -x *.o -x *.a -x *.bak zlib-1.2.3-orig/win32/Makefile.gcc zlib-1.2.3/win32/Makefile.gcc
--- zlib-1.2.3-orig/win32/Makefile.gcc	Sat Aug  2 05:49:00 2003
+++ zlib-1.2.3/win32/Makefile.gcc	Sat Dec  8 10:19:41 2007
@@ -19,25 +19,26 @@
 #
 #   make install -fmakefile.gcc
 
+SRCDIR= @SRCDIR@
+
 # Note:
 # If the platform is *not* MinGW (e.g. it is Cygwin or UWIN),
 # the DLL name should be changed from "zlib1.dll".
 
 STATICLIB = libz.a
 SHAREDLIB = zlib1.dll
-IMPLIB    = libzdll.a
+IMPLIB    = libz.dll.a
 
 #LOC = -DASMV
 #LOC = -DDEBUG -g
 
-CC = gcc
-CFLAGS = $(LOC) -O3 -Wall
+CFLAGS = $(LOC) -O3 -Wall $(GCC_ARCH_FLAGS)
 
 AS = $(CC)
 ASFLAGS = $(LOC) -Wall
 
 LD = $(CC)
-LDFLAGS = $(LOC) -s
+LDFLAGS += $(LOC) -s
 
 AR = ar
 ARFLAGS = rcs
@@ -53,6 +54,9 @@
 prefix = /usr/local
 exec_prefix = $(prefix)
 
+vpath	%.h  $(SRCDIR)
+vpath	%.c  $(SRCDIR)
+
 OBJS = adler32.o compress.o crc32.o deflate.o gzio.o infback.o \
        inffast.o inflate.o inftrees.o trees.o uncompr.o zutil.o
 OBJA =
@@ -78,10 +82,15 @@
 
 $(IMPLIB): $(SHAREDLIB)
 
-$(SHAREDLIB): win32/zlib.def $(OBJS) $(OBJA) zlibrc.o
-	dllwrap --driver-name $(CC) --def win32/zlib.def \
-	  --implib $(IMPLIB) -o $@ $(OBJS) $(OBJA) zlibrc.o
-	strip $@
+$(SHAREDLIB): $(STATICLIB)
+	$(CC) -shared $(LDFLAGS) \
+	  -o $@ \
+	  -Wl,--out-implib=$(IMPLIB) \
+	  $(SRCDIR)/win32/zlib.def \
+	  -Wl,--whole-archive $< \
+	  -Wl,--no-whole-archive \
+	  -Wl,-s
+	strip $(STRIP_FLAGS) $@
 
 example: example.o $(STATICLIB)
 	$(LD) $(LDFLAGS) -o $@ example.o $(STATICLIB)
@@ -95,8 +104,8 @@
 minigzip_d: minigzip.o $(IMPLIB)
 	$(LD) $(LDFLAGS) -o $@ minigzip.o $(IMPLIB)
 
-zlibrc.o: win32/zlib1.rc
-	$(RC) $(RCFLAGS) -o $@ win32/zlib1.rc
+zlibrc.o: $(SRCDIR)/win32/zlib1.rc
+	$(RC) $(RCFLAGS) -o $@ $<
 
 
 # INCLUDE_PATH and LIBRARY_PATH must be set.
@@ -104,18 +113,21 @@
 .PHONY: install uninstall clean
 
 install: zlib.h zconf.h $(LIB)
-	-@if not exist $(INCLUDE_PATH)/nul mkdir $(INCLUDE_PATH)
-	-@if not exist $(LIBRARY_PATH)/nul mkdir $(LIBRARY_PATH)
-	-$(INSTALL) zlib.h $(INCLUDE_PATH)
-	-$(INSTALL) zconf.h $(INCLUDE_PATH)
-	-$(INSTALL) $(STATICLIB) $(LIBRARY_PATH)
+	-mkdir -p $(INCLUDE_PATH)
+	-mkdir -p $(LIBRARY_PATH)
+	-mkdir -p $(SHAREDLIB_PATH)
+	-$(INSTALL) $(SRCDIR)/zlib.h $(INCLUDE_PATH)
+	-$(INSTALL) $(SRCDIR)/zconf.h $(INCLUDE_PATH)
+	-$(INSTALL) $(STATICLIB) $(STATICLIBRARY_PATH)
 	-$(INSTALL) $(IMPLIB) $(LIBRARY_PATH)
+	-$(INSTALL) $(SHAREDLIB) $(SHAREDLIB_PATH) 
 
 uninstall:
 	-$(RM) $(INCLUDE_PATH)/zlib.h
 	-$(RM) $(INCLUDE_PATH)/zconf.h
-	-$(RM) $(LIBRARY_PATH)/$(STATICLIB)
+	-$(RM) $(STATICLIBRARY_PATH)/$(STATICLIB)
 	-$(RM) $(LIBRARY_PATH)/$(IMPLIB)
+	-$(RM) $(SHAREDLIB_PATH)/$(SHAREDLIB)
 
 clean:
 	-$(RM) $(STATICLIB)
