diff -ur ATLAS/src/blas/level2/ATL_hbmv.c ATLAS-mod/src/blas/level2/ATL_hbmv.c
--- ATLAS/src/blas/level2/ATL_hbmv.c	2003-12-22 15:08:02.000000000 +0100
+++ ATLAS-mod/src/blas/level2/ATL_hbmv.c	2007-02-05 16:48:14.187500000 +0100
@@ -173,7 +173,7 @@
          for( k = 0; k < na; k += nb )
          {
             kb  = na - k; kb = Mmin( kb, nb );
-            ian = j + mb + k; ia = mb - K + k; ia = j + Mmax( ia, 0 );
+            ian = j + mb + k; ia = j + Mmax( mb - K + k, 0 );
             ma  = ian - ia - k; kl = ma - 1;   kl = Mmax( kl, 0 );
             ku  = K - k - 1 - kl; ku = Mmax( ku, 0 );
             gbmv0( kb, ma, kl, ku, one, A+ian*lda2, LDA, x+(ia  SHIFT), 1,
diff -ur ATLAS/src/blas/level2/ATL_sbmv.c ATLAS-mod/src/blas/level2/ATL_sbmv.c
--- ATLAS/src/blas/level2/ATL_sbmv.c	2003-12-22 15:08:03.000000000 +0100
+++ ATLAS-mod/src/blas/level2/ATL_sbmv.c	2007-02-05 16:48:50.468750000 +0100
@@ -173,7 +173,7 @@
          for( k = 0; k < na; k += nb )
          {
             kb  = na - k; kb = Mmin( kb, nb );
-            ian = j + mb + k; ia = mb - K + k; ia = j + Mmax( ia, 0 );
+            ian = j + mb + k; ia = j + Mmax( mb - K + k, 0 );
             ma  = ian - ia - k; kl = ma - 1;   kl = Mmax( kl, 0 );
             ku  = K - k - 1 - kl; ku = Mmax( ku, 0 );
             gbmv0( kb, ma, kl, ku, one, A+ian*lda2, LDA, x+(ia  SHIFT), 1,
