Index: octMakefile.in
===================================================================
RCS file: /cvs/octave/octMakefile.in,v
retrieving revision 1.232
diff -c -r1.232 octMakefile.in
*** octMakefile.in	13 Dec 2007 21:35:36 -0000	1.232
--- octMakefile.in	15 Dec 2007 18:58:54 -0000
***************
*** 97,106 ****
--- 97,120 ----
  	@$(do-subst-default-vals)
  	chmod a+rx $@
  
+ octave-config.cc: octave-config.cc.in Makeconf octMakefile
+ 	@$(do-subst-default-vals)
+ 
+ octave-config$(EXEEXT): octave-config.o
+ 	$(LD_CXX) $(CPPFLAGS) $(ALL_CXXFLAGS) $(RDYNAMIC_FLAG) \
+ 	$(ALL_LDFLAGS) -o $@ octave-config.o
+ 
  mkoctfile: mkoctfile.in Makeconf octMakefile $(top_srcdir)/src/version.h
  	@$(do-subst-config-vals)
  	chmod a+rx $@
  
+ mkoctfile.cc: mkoctfile.cc.in Makeconf octMakefile
+ 	@$(do-subst-config-vals)
+ 
+ mkoctfile$(EXEEXT): mkoctfile.o
+ 	$(LD_CXX) $(CPPFLAGS) $(ALL_CXXFLAGS) $(RDYNAMIC_FLAG) \
+ 	$(ALL_LDFLAGS) -o $@ mkoctfile.o
+ 
  run-octave: run-octave.in Makeconf octMakefile
  	@$(do-subst-script-vals)
  	chmod a+rx "$@"
