diff -urN -x '*.exe' -x '*.dll' -x '*.o' -x '*.a' -x '*.bak' netcdf-4.0-orig/libsrc/netcdf.h netcdf-4.0/libsrc/netcdf.h
--- netcdf-4.0-orig/libsrc/netcdf.h	2008-03-28 21:50:28 +0100
+++ netcdf-4.0/libsrc/netcdf.h	2009-02-14 17:55:36 +0100
@@ -223,10 +223,17 @@
 #   define MSC_EXTRA __declspec(dllimport)
 #  endif
 #include <io.h>
+#if defined(_MSC_VER)
 #define lseek _lseeki64
 #define off_t __int64
 #define stat __stat64
 #define fstat _fstat64
+#elif defined(__MINGW32__)
+#define lseek _lseeki64
+#define off_t __int64
+#define stat _stati64
+#define fstat _fstati64
+#endif  /* __MINGW32__ */
 #else
 #define MSC_EXTRA
 #endif	/* defined(DLL_NETCDF) */
