diff -urN -x *.exe -x *.dll -x *.o -x *.a -x *.bak expat-2.0.1-orig/configure expat-2.0.1/configure
--- expat-2.0.1-orig/configure	Tue Jun  5 18:19:38 2007
+++ expat-2.0.1/configure	Mon Apr 21 11:27:39 2008
@@ -7954,7 +7954,7 @@
       ;;
     mingw*)
       # MinGW DLLs use traditional 'lib' prefix
-      soname_spec='${libname}`echo ${release} | $SED -e 's/[.]/-/g'`${versuffix}${shared_ext}'
+      soname_spec='`echo ${libname} | sed -e 's/^lib//'`${shared_ext}'
       sys_lib_search_path_spec=`$CC -print-search-dirs | grep "^libraries:" | $SED -e "s/^libraries://" -e "s,=/,/,g"`
       if echo "$sys_lib_search_path_spec" | grep ';[c-zC-Z]:/' >/dev/null; then
         # It is most probably a Windows format PATH printed by
