diff -urN -x '*.exe' -x '*.dll' -x '*.o' -x '*.a' -x '*.bak' -x '*.def' cblas-orig/Makefile cblas/Makefile
--- cblas-orig/Makefile	1999-03-15 21:26:42 +0100
+++ cblas/Makefile	2007-12-08 21:43:49 +0100
@@ -1,4 +1,4 @@
-dlvl = ./.
+dlvl = @SRCDIR@
 include $(dlvl)/Makefile.in
 help:
 	@ echo "Make sure you are using correct Makefile.in for your system."
diff -urN -x '*.exe' -x '*.dll' -x '*.o' -x '*.a' -x '*.bak' -x '*.def' cblas-orig/makefile.in cblas/makefile.in
--- cblas-orig/makefile.in	1970-01-01 01:00:00 +0100
+++ cblas/makefile.in	2008-08-24 14:29:00 +0200
@@ -0,0 +1,54 @@
+#
+# Makefile.LINUX
+#
+#
+# If you compile, change the name to Makefile.in.
+#
+#
+ 
+#-----------------------------------------------------------------------------
+# Shell
+#-----------------------------------------------------------------------------
+
+SHELL = /bin/sh
+
+#-----------------------------------------------------------------------------
+# Platform
+#-----------------------------------------------------------------------------
+
+PLAT = MINGW32
+
+#-----------------------------------------------------------------------------
+# Libraries and includes
+#-----------------------------------------------------------------------------
+ 
+BLLIB = -lblas
+CBDIR = $(TOPDIR)/$(SRCDIR)
+CBLIBDIR = ../lib
+CBLIB = $(CBLIBDIR)/libcblas.a
+CBLIBDLL=$(CBLIBDIR)/cblas.dll
+IMPLIB=$(CBLIBDIR)/libcblas.dll.a
+DEFFILE=$(CBLIB:.a=.def)
+
+#-----------------------------------------------------------------------------
+# Compilers
+#-----------------------------------------------------------------------------
+
+#CC = gcc
+FC = $(F77)
+LOADER = $(FC)
+
+#-----------------------------------------------------------------------------
+# Flags for Compilers
+#-----------------------------------------------------------------------------
+
+CFLAGS = -O3 -DADD_ $(GCC_ARCH_FLAGS)
+FFLAGS = -O3  
+
+#-----------------------------------------------------------------------------
+# Archive programs and flags
+#-----------------------------------------------------------------------------
+
+ARCH = ar
+ARCHFLAGS = r
+RANLIB = echo
diff -urN -x '*.exe' -x '*.dll' -x '*.o' -x '*.a' -x '*.bak' -x '*.def' cblas-orig/src/Makefile cblas/src/Makefile
--- cblas-orig/src/Makefile	1999-03-15 21:11:03 +0100
+++ cblas/src/Makefile	2008-08-24 14:45:36 +0200
@@ -1,9 +1,12 @@
 # This Makefile compiles the CBLAS routines
 #
-dlvl = ../.
+dlvl = @SRCDIR@
 include $(dlvl)/Makefile.in
 INC = -I$(CBDIR)/src
 
+vpath %.c $(dlvl)/src
+vpath %.f $(dlvl)/src
+
 #
 # Erase all object and archive files
 #
@@ -230,15 +233,24 @@
 	$(ARCH) $(ARCHFLAGS) $(CBLIB) $(alev3) 
 	$(RANLIB) $(CBLIB)
 
+$(DEFFILE):
+	echo EXPORTS>$@
+	sed -ne "s/^\(void\|float\|CBLAS_INDEX\)[ \t]*\(cblas_[^(]\+\)(.*$$/\2/p" $(dlvl)/src/cblas_*.c >> $@
+
 # All levels and precisions
-all: $(alev)  
+all: $(alev) $(DEFFILE) 
 	$(ARCH) $(ARCHFLAGS)  $(CBLIB) $(alev) 
 	$(RANLIB) $(CBLIB)
+	$(F77) -shared $(FLDFLAGS) -o $(CBLIBDLL) $(alev) $(DEFFILE)\
+	  -Wl,--out-implib=$(IMPLIB) \
+	  -Wl,--allow-multiple-definition \
+	  -Wl,--output-def=cblas_out.def \
+	  -Wl,--no-whole-archive $(BLLIB)
 
 
 .SUFFIXES: .o .c .f
 
 .c.o:
-	$(CC) $(CFLAGS) $(INC) -c $*.c
+	$(CC) $(CFLAGS) $(INC) -c $<
 .f.o:
-	$(FC) $(FFLAGS) -c $*.f
+	$(FC) $(FFLAGS) -c $<
