diff -urN -x '*.exe' -x '*.dll' -x '*.o' -x '*.a' -x '*.bak' -x '~' -x .hg -x '*.orig' pango-1.26.1-orig/pango.pc.in pango-1.26.1/pango.pc.in
--- pango-1.26.1-orig/pango.pc.in	2009-02-07 00:49:34 +0100
+++ pango-1.26.1/pango.pc.in	2009-12-14 22:01:49 +0100
@@ -10,4 +10,4 @@
 Version: @VERSION@
 Requires: glib-2.0,gobject-2.0,gmodule-no-export-2.0
 Libs: -L${libdir} -lpango-@PANGO_API_VERSION@ @PKGCONFIG_MATH_LIBS@
-Cflags: -I${includedir}/pango-1.0
+Cflags: -I${includedir}/pango
diff -urN -x '*.exe' -x '*.dll' -x '*.o' -x '*.a' -x '*.bak' -x '~' -x .hg -x '*.orig' pango-1.26.1-orig/pangocairo.pc.in pango-1.26.1/pangocairo.pc.in
--- pango-1.26.1-orig/pangocairo.pc.in	2009-02-07 00:49:34 +0100
+++ pango-1.26.1/pangocairo.pc.in	2009-12-14 22:01:49 +0100
@@ -8,4 +8,4 @@
 Version: @VERSION@
 Requires: pango cairo @PKGCONFIG_CAIRO_REQUIRES@
 Libs: -L${libdir} -lpangocairo-@PANGO_API_VERSION@
-Cflags: -I${includedir}/pango-1.0
+Cflags: -I${includedir}/pango
diff -urN -x '*.exe' -x '*.dll' -x '*.o' -x '*.a' -x '*.bak' -x '~' -x .hg -x '*.orig' pango-1.26.1-orig/pangoft2.pc.in pango-1.26.1/pangoft2.pc.in
--- pango-1.26.1-orig/pangoft2.pc.in	2009-02-07 00:49:34 +0100
+++ pango-1.26.1/pangoft2.pc.in	2009-12-14 22:01:49 +0100
@@ -8,4 +8,4 @@
 Version: @VERSION@
 Requires: pango freetype2 fontconfig
 Libs: -L${libdir} -lpangoft2-@PANGO_API_VERSION@
-Cflags: -I${includedir}/pango-1.0
+Cflags: -I${includedir}/pango
diff -urN -x '*.exe' -x '*.dll' -x '*.o' -x '*.a' -x '*.bak' -x '~' -x .hg -x '*.orig' pango-1.26.1-orig/pangowin32.pc.in pango-1.26.1/pangowin32.pc.in
--- pango-1.26.1-orig/pangowin32.pc.in	2009-02-07 00:49:34 +0100
+++ pango-1.26.1/pangowin32.pc.in	2009-12-14 22:01:49 +0100
@@ -8,4 +8,4 @@
 Version: @VERSION@
 Requires: pango
 Libs: -L${libdir} -lpangowin32-@PANGO_API_VERSION@ @WIN32_LIBS@
-Cflags: -I${includedir}/pango-1.0
+Cflags: -I${includedir}/pango
