diff -urN -x '*.exe' -x '*.dll' -x '*.o' -x '*.a' -x '*.bak' pango-1.22.2-orig/pango/Makefile.in pango-1.22.2/pango/Makefile.in
--- pango-1.22.2-orig/pango/Makefile.in	2008-10-28 20:32:10 +0100
+++ pango-1.22.2/pango/Makefile.in	2008-11-10 21:00:04 +0100
@@ -1398,7 +1398,7 @@
 	uninstall-pangoincludeHEADERS
 
 pango-win32-res.o: pango.rc
-	$(WINDRES) $< $@
+	$(WINDRES) --preprocessor=$(CPP) $< $@
 
 pango-enum-types.h: s-enum-types-h
 	@true
@@ -1430,9 +1430,9 @@
 pango-features.h:
 	cd $(top_builddir) && ./config.status pango/pango-features.h
 pangoft2-win32-res.o: pangoft2.rc
-	$(WINDRES) $< $@
+	$(WINDRES) --preprocessor=$(CPP) $< $@
 pangowin32-win32-res.o: pangowin32.rc
-	$(WINDRES) $< $@
+	$(WINDRES) --preprocessor=$(CPP) $< $@
 
 # ------------------- win32 build stuff -------------------
 
