diff -urN -x *.exe -x *.dll -x *.o -x *.a -x *.bak jpeg-6b-orig/makefile.cfg jpeg-6b/makefile.cfg
--- jpeg-6b-orig/makefile.cfg	Sat Mar 21 20:08:57 1998
+++ jpeg-6b/makefile.cfg	Tue Jan  1 16:34:29 2008
@@ -133,7 +133,15 @@
 TROBJECTS= jpegtran.$(O) rdswitch.$(O) cdjpeg.$(O) transupp.$(O)
 
 
-all: @A2K_DEPS@ libjpeg.$(A) cjpeg djpeg jpegtran rdjpgcom wrjpgcom
+all: @A2K_DEPS@ libjpeg.$(A) cjpeg djpeg jpegtran rdjpgcom wrjpgcom libjpeg.dll
+
+libjpeg.dll : libjpeg.a
+	$(CC) -shared $(LDFLAGS) -o $@ \
+	-Wl,--export-all-symbols \
+	-Wl,--whole-archive $< \
+	-Wl,--out-implib=$(basename $<).dll.a \
+	-Wl,--no-whole-archive
+	strip $(STRIP_FLAGS) $@
 
 # Special compilation rules to support ansi2knr and libtool.
 .SUFFIXES: .lo .la
