diff -urN -x '*.exe' -x '*.dll' -x '*.o' -x '*.a' -x '*.bak' gmp-4.2.2-orig/configure gmp-4.2.2/configure
--- gmp-4.2.2-orig/configure	2007-09-10 01:10:47 +0200
+++ gmp-4.2.2/configure	2008-07-31 21:46:06 +0200
@@ -13750,7 +13750,7 @@
       ;;
     mingw*)
       # MinGW DLLs use traditional 'lib' prefix
-      soname_spec='${libname}`echo ${release} | $SED -e 's/[.]/-/g'`${versuffix}${shared_ext}'
+      soname_spec='${name}${shared_ext}'
       sys_lib_search_path_spec=`$CC -print-search-dirs | grep "^libraries:" | $SED -e "s/^libraries://" -e "s,=/,/,g"`
       if echo "$sys_lib_search_path_spec" | grep ';[c-zC-Z]:/' >/dev/null; then
         # It is most probably a Windows format PATH printed by
