diff -urN image-1.0.10-orig//src/jpgread.cc image-1.0.10//src/jpgread.cc
--- image-1.0.10-orig//src/jpgread.cc	2009-06-07 12:45:04 +0200
+++ image-1.0.10//src/jpgread.cc	2009-10-26 18:59:15 +0100
@@ -46,6 +46,11 @@
 extern "C" {
 #endif
 
+#if defined (__MINGW32__)
+#define HAVE_BOOLEAN
+#define boolean int
+#endif
+
 #include "jpeglib.h"
 
 #ifdef __cplusplus
diff -urN image-1.0.10-orig//src/jpgwrite.cc image-1.0.10//src/jpgwrite.cc
--- image-1.0.10-orig//src/jpgwrite.cc	2009-06-07 12:45:04 +0200
+++ image-1.0.10//src/jpgwrite.cc	2009-10-26 19:10:40 +0100
@@ -15,6 +15,11 @@
 extern "C" {
 #endif
 
+#if defined (__MINGW32__)
+#define HAVE_BOOLEAN
+#define boolean int
+#endif
+
 #include "jpeglib.h"
 
 #ifdef __cplusplus
