Index: java-1.2.7/PKG_ADD
===================================================================
--- java-1.2.7/PKG_ADD	(revision 7108)
+++ java-1.2.7/PKG_ADD	(working copy)
@@ -9,3 +9,31 @@
 autoload ("java_unsigned_conversion", fullfile (fileparts (mfilename ("fullpath")), "__java__.oct"));
 autoload ("java_debug", fullfile (fileparts (mfilename ("fullpath")), "__java__.oct"));
 atexit ("java_exit");
+
+if (file_in_path (EXEC_PATH, "dom4j-1.6.1.jar"))
+    javaaddpath (file_in_path (EXEC_PATH, "dom4j-1.6.1.jar"));
+endif
+if (file_in_path (EXEC_PATH, "jOpenDocument-1.2b2.jar"))
+    javaaddpath (file_in_path (EXEC_PATH, "jOpenDocument-1.2b2.jar"));
+endif
+if (file_in_path (EXEC_PATH, "jxl.jar"))
+    javaaddpath (file_in_path (EXEC_PATH, "jxl.jar"));
+endif
+if (file_in_path (EXEC_PATH, "odfdom.jar"))
+    javaaddpath (file_in_path (EXEC_PATH, "odfdom.jar"));
+endif
+if (file_in_path (EXEC_PATH, "poi-3.6-20091214.jar"))
+    javaaddpath (file_in_path (EXEC_PATH, "poi-3.6-20091214.jar"));
+endif
+if (file_in_path (EXEC_PATH, "poi-ooxml-3.6-20091214.jar"))
+    javaaddpath (file_in_path (EXEC_PATH, "poi-ooxml-3.6-20091214.jar"));
+endif
+if (file_in_path (EXEC_PATH, "poi-ooxml-schemas-3.6-20091214.jar"))
+    javaaddpath (file_in_path (EXEC_PATH, "poi-ooxml-schemas-3.6-20091214.jar"));
+endif
+if (file_in_path (EXEC_PATH, "xbean.jar"))
+    javaaddpath (file_in_path (EXEC_PATH, "xbean.jar"));
+endif
+if (file_in_path (EXEC_PATH, "xercesImpl.jar"))
+    javaaddpath (file_in_path (EXEC_PATH, "xercesImpl.jar"));
+endif
Index: java-1.2.7/src/__java__.cc
===================================================================
--- java-1.2.7/src/__java__.cc	(revision 7108)
+++ java-1.2.7/src/__java__.cc	(working copy)
@@ -765,7 +765,7 @@
       if (jni_env->IsInstanceOf (jobj, cls))
         {
           jmethodID m = jni_env->GetMethodID (cls, "booleanValue", "()Z");
-          retval = jni_env->CallBooleanMethod (jobj, m);
+          retval = (jni_env->CallBooleanMethod (jobj, m) ? true : false);
         }
     }
 
@@ -958,6 +958,15 @@
       std::string s = val.string_value ();
       jobj = jni_env->NewStringUTF (s.c_str ());
       jcls = jni_env->GetObjectClass (jobj);
+    }
+  else if (val.is_bool_scalar ())
+    {
+      bool bval = val.bool_value ();
+      jclass_ref bcls (jni_env, jni_env->FindClass ("java/lang/Boolean"));
+      jfieldID fid = jni_env->GetStaticFieldID (bcls, "TYPE", "Ljava/lang/Class;");
+      jmethodID mid = jni_env->GetMethodID (bcls, "<init>", "(Z)V");
+      jcls = reinterpret_cast<jclass> (jni_env->GetStaticObjectField(bcls, fid));
+      jobj = jni_env->NewObject (bcls, mid, bval);
     }
   else if (val.is_real_scalar ())
     {
