diff -urN -x '*.exe' -x '*.dll' -x '*.o' -x '*.a' -x '*.bak' pcre-7.8-orig/configure pcre-7.8/configure
--- pcre-7.8-orig/configure	2008-09-05 17:58:01 +0200
+++ pcre-7.8/configure	2008-10-31 08:32:16 +0100
@@ -9617,7 +9617,7 @@
       ;;
     mingw*)
       # MinGW DLLs use traditional 'lib' prefix
-      soname_spec='${libname}`echo ${release} | $SED -e 's/[.]/-/g'`${versuffix}${shared_ext}'
+      soname_spec='`echo ${libname} | sed -e 's/^lib//'``echo ${release} | $SED -e 's/[.]/-/g'`${versuffix}${shared_ext}'
       sys_lib_search_path_spec=`$CC -print-search-dirs | grep "^libraries:" | $SED -e "s/^libraries://" -e "s,=/,/,g"`
       if echo "$sys_lib_search_path_spec" | grep ';[c-zC-Z]:/' >/dev/null; then
         # It is most probably a Windows format PATH printed by
@@ -22835,13 +22835,13 @@
 # (Note: The libpcre*_version bits are m4 variables, assigned above)
 
 EXTRA_LIBPCRE_LDFLAGS="$EXTRA_LIBPCRE_LDFLAGS \
-                       $NO_UNDEFINED -version-info 0:1:0"
+                       $NO_UNDEFINED -version-info 7:8:0"
 
 EXTRA_LIBPCREPOSIX_LDFLAGS="$EXTRA_LIBPCREPOSIX_LDFLAGS \
-                            $NO_UNDEFINED -version-info 0:0:0"
+                            $NO_UNDEFINED -version-info 7:8:0"
 
 EXTRA_LIBPCRECPP_LDFLAGS="$EXTRA_LIBPCRECPP_LDFLAGS \
-                          $NO_UNDEFINED -version-info 0:0:0 \
+                          $NO_UNDEFINED -version-info 7:8:0 \
                           $EXPORT_ALL_SYMBOLS"
 
 
